--- common/sysutils.c.dist	2016-04-09 18:54:25.000000000 +0100
+++ common/sysutils.c	2016-04-09 19:14:07.000000000 +0100
@@ -154,8 +154,11 @@ get_session_marker( size_t *rlen )
          * take place.  Of course, we can add just more random but
          * than we need the random generator even for verification
          * tasks - which does not make sense. */
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wuninitialized"
         a = aa ^ (ulong)getpid();
         b = bb ^ (ulong)time(NULL);
+#pragma clang diagnostic pop
         memcpy( marker, &a, SIZEOF_UNSIGNED_LONG );
         memcpy( marker+SIZEOF_UNSIGNED_LONG, &b, SIZEOF_UNSIGNED_LONG );
     }
--- sm/certreqgen-ui.c.dist	2016-04-09 19:04:08.000000000 +0100
+++ sm/certreqgen-ui.c	2016-04-09 19:07:46.000000000 +0100
@@ -143,7 +143,6 @@ gpgsm_gencertreq_tty (ctrl_t ctrl, FILE
   char *subject_name;
   membuf_t mb_email, mb_dns, mb_uri, mb_result;
   char *result = NULL;
-  int i;
   const char *s, *s2;
 
   answer = NULL;
@@ -369,7 +368,7 @@ gpgsm_gencertreq_tty (ctrl_t ctrl, FILE
     goto mem_error;
 
   tty_printf (_("Parameters to be used for the certificate request:\n"));
-  for (s=result; (s2 = strchr (s, '\n')); s = s2+1, i++)
+  for (s=result; (s2 = strchr (s, '\n')); s = s2+1)
     tty_printf ("    %.*s\n", (int)(s2-s), s);
   tty_printf ("\n");
 
--- tools/symcryptrun.c.dist	2016-04-09 19:59:39.000000000 +0100
+++ tools/symcryptrun.c	2016-04-09 20:02:28.000000000 +0100
@@ -531,7 +531,10 @@ confucius_process (int mode, char *infil
       return 1;
     }
 
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wimplicit-function-declaration"
   if (openpty (&master, &slave, NULL, NULL, NULL) == -1)
+#pragma clang diagnostic pop
     {
       log_error (_("could not create pty: %s\n"), strerror (errno));
       close (cstderr[0]);
@@ -566,7 +569,10 @@ confucius_process (int mode, char *infil
       close (cstderr[0]);
 
       /* Change controlling terminal.  */
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wimplicit-function-declaration"
       if (login_tty (slave))
+#pragma clang diagnostic pop
 	{
 	  /* It's too early to output a debug message.  */
 	  _exit (1);
