--- a/include/ntop_defines.h
+++ b/include/ntop_defines.h
@@ -121,9 +121,9 @@
 #define MAX_NUM_VIEW_INTERFACES   8
 
 #define LIMITED_NUM_INTERFACES    32
-#define LIMITED_NUM_HOST_POOLS    4 /* 3 pools plus the NO_HOST_POOL_ID */
-#define LIMITED_NUM_PROFILES      16
-#define LIMITED_NUM_POOL_MEMBERS  8
+#define LIMITED_NUM_HOST_POOLS    128 /* 4 */ /* 3 pools plus the NO_HOST_POOL_ID */
+#define LIMITED_NUM_PROFILES      128 /* 16 */
+#define LIMITED_NUM_POOL_MEMBERS  256 /* 8 */
 #ifndef NTOPNG_PRO
 #define MAX_NUM_DEFINED_INTERFACES  LIMITED_NUM_INTERFACES
 #define MAX_NUM_HOST_POOLS          LIMITED_NUM_HOST_POOLS
