--- a/src/Utils.cpp
+++ b/src/Utils.cpp
@@ -373,7 +373,7 @@ int Utils::dropPrivileges() {
     ntop->getTrace()->traceEvent(TRACE_WARNING, "Unable to locate user %s", username);
     return -1;
   }
-  umask(0);
+  umask(S_IWGRP | S_IROTH | S_IWOTH);
 #endif
   return 0;
 }
--- a/src/Ntop.cpp
+++ b/src/Ntop.cpp
@@ -90,7 +90,7 @@ Ntop::Ntop(char *appName) {
   /* Folder will be created lazily, avoid creating it now */
   snprintf(working_dir, sizeof(working_dir), "%s/ntopng", CONST_DEFAULT_WRITABLE_DIR);
 
-  umask(0);
+  umask(S_IWGRP | S_IROTH | S_IWOTH);
 
   if(getcwd(startup_dir, sizeof(startup_dir)) == NULL)
     ntop->getTrace()->traceEvent(TRACE_ERROR,
@@ -1321,7 +1321,7 @@ void Ntop::daemonize() {
       /*
        * clear any inherited file mode creation mask
        */
-      umask(0);
+      umask(S_IWGRP | S_IROTH | S_IWOTH);
 
       /*
        * Use line buffered stdout
