--- a/configure.seed
+++ b/configure.seed
@@ -53,8 +53,6 @@
 # On CentOS 6 `git rev-list HEAD --count` does not work
 #
 #
-REVISION=`git log --pretty=oneline | wc -l`
-
 if test -d "/usr/local/include"; then
 CFLAGS="${CFLAGS} -I/usr/local/include"
 CPPFLAGS="${CPPFLAGS} -I/usr/local/include"
@@ -82,101 +80,6 @@
 
 SHORT_MACHINE=`uname -m | cut -b1-3`
 
-GIT_RELEASE="@GIT_RELEASE@"
-GIT_DATE="@GIT_DATE@"
-GIT_BRANCH="@GIT_BRANCH@"
-PRO_GIT_RELEASE="@PRO_GIT_RELEASE@"
-PRO_GIT_DATE="@PRO_GIT_DATE@"
-
-dnl start: nDPI handling
-AC_ARG_WITH(ndpi-static-lib,
-   [  --with-ndpi-static-lib=DIR       Search the specified directory for nDPI static library],
-   [
-      if test -d $withval ; then :
-         AC_MSG_CHECKING(your own nDPI library)
-         if test -r $withval/libndpi.a; then
-            AC_MSG_RESULT(ok)
-            NDPI_CUST_LIB=$withval/libndpi.a
-         else
-            AC_MSG_RESULT(no)
-            AC_MSG_ERROR(ERROR: Unable to find custom nDPI static library in $withval)
-         fi
-      else
-         AC_MSG_ERROR(ERROR: Can't open custom nDPI static library directory $withval)
-      fi
-   ])
-AC_ARG_WITH(ndpi-includes,
-   [  --with-ndpi-includes=DIR         Search the specified directory for nDPI header files],
-   [
-      if test ! -d $withval ; then :
-         AC_MSG_ERROR(ERROR: Can't open custom nDPI header files directory $withval)
-      fi
-      if test ! -r $withval/ndpi_main.h ; then :
-         AC_MSG_ERROR(ERROR: Can't read ndpi_main.h in custom nDPI header files directory $withval)
-      fi
-      NDPI_CUST_INC="-I$withval -I$withval/../lib/third_party/include"
-   ])
-
-EBPF_HOME="../ebpf/libebpfflow/"
-AC_MSG_CHECKING(for ebpfflowlib)
-if test -f "$EBPF_HOME/libebpfflow.a" ; then :
-       AC_MSG_RESULT(found $EBPF_HOME)
-       PRO_INCS="${PRO_INCS} -I ${EBPF_HOME}"
-       PRO_LIBS="${PRO_LIBS} -L${EBPF_HOME} -lebpfflow -lbcc"
-       AC_DEFINE_UNQUOTED(HAVE_EBPF, 1, [Building with eBPF support])
-else
-       AC_MSG_RESULT(not found)
-fi
-
-PKG_CHECK_MODULES([NDPI], [libndpi >= 2.0], [
-   NDPI_INC=`echo $NDPI_CFLAGS | sed -e "s/[ ]*$//"`/libndpi
-   # Use static libndpi library as building against the dynamic library fails
-   NDPI_LIB="-Wl,-Bstatic $NDPI_LIBS -Wl,-Bdynamic"
-   NDPI_LIB_DEP=
-   ], [
-      AC_MSG_CHECKING(for nDPI source)
-      if test -d "./nDPI" ; then :
-         NDPI_HOME=./nDPI
-      elif test -d "../nDPI" ; then :
-         NDPI_HOME=../nDPI
-      elif test -d "$HOME/nDPI" ; then :
-         NDPI_HOME=$HOME/nDPI
-      fi
-
-      if test ! -z "$NDPI_HOME" ; then :
-         AC_MSG_RESULT(found in $NDPI_HOME)
-         NDPI_LIB=$NDPI_HOME/src/lib/libndpi.a
-         AC_MSG_CHECKING(for $NDPI_LIB)
-         if test -r $NDPI_LIB ; then :
-            AC_MSG_RESULT(found $NDPI_LIB)
-         else
-            AC_MSG_RESULT(not found $NDPI_LIB: compiling)
-            cd $NDPI_HOME; ./autogen.sh; make; cd -
-         fi
-         NDPI_INC="-I$NDPI_HOME/src/include -I$NDPI_HOME/src/lib/third_party/include"
-         NDPI_LIB=$NDPI_HOME/src/lib/libndpi.a
-         NDPI_LIB_DEP=$NDPI_LIB
-      else
-         AC_MSG_RESULT(not found)
-      fi
-   ])
-
-if test ! -z "$NDPI_CUST_INC" ; then :
-   NDPI_INC=$NDPI_CUST_INC
-fi
-
-if test ! -z "$NDPI_CUST_LIB" ; then :
-   NDPI_LIB=$NDPI_CUST_LIB
-fi
-
-if test -z "$NDPI_INC" || test -z "$NDPI_LIB" ; then :
-   echo "Could not find nDPI"
-   echo "Please do cd ..; git clone https://github.com/ntop/nDPI.git; cd nDPI; ./autogen.sh; make; cd ../ntopng"
-   echo "and try again"
-   exit 1
-fi
-dnl finish: nDPI handling
-
 if test -z `pkg-config --exists libssl`; then
  AC_DEFINE_UNQUOTED(NO_SSL_DL, 1, [has openssl])
  SSL_INC="`pkg-config --cflags libssl` -I/usr/include/openssl"
