--- a/include/ntop_defines.h
+++ b/include/ntop_defines.h
@@ -143,9 +143,9 @@
 #define MAX_NUM_VIEW_INTERFACES   8
 
 #define LIMITED_NUM_INTERFACES    32
-#define LIMITED_NUM_HOST_POOLS    4 /* 3 pools plus the NO_HOST_POOL_ID */
-#define LIMITED_NUM_PROFILES      16
-#define LIMITED_NUM_POOL_MEMBERS  8
+#define LIMITED_NUM_HOST_POOLS    128 /* 4 */ /* 3 pools plus the NO_HOST_POOL_ID */
+#define LIMITED_NUM_PROFILES      128 /* 16 */
+#define LIMITED_NUM_POOL_MEMBERS  256 /* 8 */
 #define UNLIMITED_NUM_INTERFACES  128
 #define UNLIMITED_NUM_HOST_POOLS  128
 #define UNLIMITED_NUM_PROFILES    128
