diff --git a/sound/soc/cix/card-utils.c b/sound/soc/cix/card-utils.c
--- a/sound/soc/cix/card-utils.c
+++ b/sound/soc/cix/card-utils.c
@@ -167,6 +167,10 @@ static int dai_set_sysclk(struct snd_pcm_substream *substream,
 		} else if (sample_rate == 192000) {
 			mclk_fs = 256;
 			mclk_div = 0;
+		} else {
+			dev_err(rtd->dev, "Unsupported sample rate %u for 8kHz clock family\n",
+				sample_rate);
+			return -EINVAL;
 		}
 
 		mclk_parent = link_info->clks[AUDIO_CLK0];
@@ -186,6 +190,10 @@ static int dai_set_sysclk(struct snd_pcm_substream *substream,
 		} else if (sample_rate == 176400) {
 			mclk_fs = 256;
 			mclk_div = 0;
+		} else {
+			dev_err(rtd->dev, "Unsupported sample rate %u for 11.025kHz clock family\n",
+				sample_rate);
+			return -EINVAL;
 		}
 
 		mclk_parent = link_info->clks[AUDIO_CLK2];
