diff --git a/drivers/gpu/drm/panthor/panthor_gem.c b/drivers/gpu/drm/panthor/panthor_gem.c
--- a/drivers/gpu/drm/panthor/panthor_gem.c
+++ b/drivers/gpu/drm/panthor/panthor_gem.c
@@ -80,8 +80,6 @@ static void panthor_gem_debugfs_bo_init(struct panthor_gem_object *bo) {}
 static bool
 should_map_wc(struct panthor_gem_object *bo, struct panthor_vm *exclusive_vm)
 {
-	struct panthor_device *ptdev = container_of(bo->base.base.dev, struct panthor_device, base);
-
 	/* We can't do uncached mappings if the device is coherent,
 	 * because the zeroing done by the shmem layer at page allocation
 	 * time happens on a cached mapping which isn't CPU-flushed (at least
