diff --git a/drivers/misc/armchina-npu/sky1/sky1.c b/drivers/misc/armchina-npu/sky1/sky1.c
--- a/drivers/misc/armchina-npu/sky1/sky1.c
+++ b/drivers/misc/armchina-npu/sky1/sky1.c
@@ -435,11 +435,6 @@ static int sky1_npu_probe(struct platform_device *p_dev)
 npu_probe_failed:
 	pm_runtime_put(&p_dev->dev);
 
-    if (has_acpi_companion(&p_dev->dev)) {
-		for (i = 0; i < CIX_NPU_PD_NUM; i++) {
-			pm_runtime_disable(cix_aipu_priv->pd_core[i]);
-		}
-	}
 	pm_runtime_disable(&p_dev->dev);
 
 #ifdef CONFIG_ENABLE_DEVFREQ
@@ -448,6 +443,12 @@ static int sky1_npu_probe(struct platform_device *p_dev)
 
 devfreq_init_failed:
 	for (i=0; i < CIX_NPU_PD_NUM; i++) {
+		if (!cix_aipu_priv->pd_core[i])
+			continue;
+
+		if (has_acpi_companion(&p_dev->dev))
+			pm_runtime_disable(cix_aipu_priv->pd_core[i]);
+
 		dev_pm_domain_detach(cix_aipu_priv->pd_core[i], true);
 	}
 
