--- util/domd.dist	2014-03-31 19:56:39.000000000 +0100
+++ util/domd	2014-03-31 19:57:30.000000000 +0100
@@ -2,6 +2,8 @@
 # Do a makedepend, only leave out the standard headers
 # Written by Ben Laurie <ben@algroup.co.uk> 19 Jan 1999
 
+PERL="${PERL:-perl}"
+
 TOP=$1
 shift
 if [ "$1" = "-MD" ]; then
@@ -14,7 +16,7 @@
 cp Makefile Makefile.save
 # fake the presence of Kerberos
 touch $TOP/krb5.h
-if expr "$MAKEDEPEND" : '.*gcc$' > /dev/null; then
+if expr "$MAKEDEPEND" : '.*gcc$' > /dev/null || expr "$MAKEDEPEND" : '.*clang$' > /dev/null; then
     args=""
     while [ $# -gt 0 ]; do
 	if [ "$1" != "--" ]; then args="$args $1"; fi
