--- a/config/profile.sh.in
+++ b/config/profile.sh.in
@@ -5,13 +5,33 @@
 # such as $DEBUGINFOD_MAXSIZE, $DEBUGINFOD_MAXTIME, $DEBUGINFOD_PROGRESS.
 
 prefix="@prefix@"
-if [ -z "$DEBUGINFOD_URLS" ]; then
-    DEBUGINFOD_URLS=$(find "@sysconfdir@/debuginfod" -name "*.urls" -print0 2>/dev/null | xargs -0 cat 2>/dev/null | tr '\n' ' ' || :)
-    [ -n "$DEBUGINFOD_URLS" ] && export DEBUGINFOD_URLS || unset DEBUGINFOD_URLS
+if [ -z "${DEBUGINFOD_URLS:-}" ]; then
+    DEBUGINFOD_URLS="$(
+			find "@sysconfdir@/debuginfod" \
+					-name "*.urls" \
+					-print0 2>/dev/null |
+				xargs -0 cat 2>/dev/null |
+				tr '\n' ' '
+		)" || :
+    if [ -n "${DEBUGINFOD_URLS:-}" ]; then
+		export DEBUGINFOD_URLS
+	else
+		unset DEBUGINFOD_URLS
+	fi
 fi
 
-if [ -z "$DEBUGINFOD_IMA_CERT_PATH" ]; then
-    DEBUGINFOD_IMA_CERT_PATH=$(find "@sysconfdir@/debuginfod" -name "*.certpath" -print0 2>/dev/null | xargs -0 cat 2>/dev/null | tr '\n' ':' || :)
-    [ -n "$DEBUGINFOD_IMA_CERT_PATH" ] && export DEBUGINFOD_IMA_CERT_PATH || unset DEBUGINFOD_IMA_CERT_PATH
+if [ -z "${DEBUGINFOD_IMA_CERT_PATH:-}" ]; then
+    DEBUGINFOD_IMA_CERT_PATH="$(
+			find "@sysconfdir@/debuginfod" \
+					-name "*.certpath" \
+					-print0 2>/dev/null |
+				xargs -0 cat 2>/dev/null |
+				tr '\n' ':'
+		)" || :
+    if [ -n "${DEBUGINFOD_IMA_CERT_PATH:-}" ]; then
+		export DEBUGINFOD_IMA_CERT_PATH
+	else
+		unset DEBUGINFOD_IMA_CERT_PATH
+	fi
 fi
 unset prefix
