--- src/protocols/imap.c.dist	2015-01-23 00:42:42.863654426 +0000
+++ src/protocols/imap.c	2015-01-23 01:07:35.984251190 +0000
@@ -40,6 +40,7 @@ int check_imap(Socket_T socket) {
         char buf[STRLEN];
         const char *ok = "* OK";
         const char *bye = "* BYE";
+        //const char *success = "Success";
 
         ASSERT(socket);
 
@@ -59,15 +60,30 @@ int check_imap(Socket_T socket) {
                 socket_setError(socket, "IMAP: logout command error -- %s", STRERROR);
                 return FALSE;
         }
+
+        // Dovecot appears to output a '* BYE Logging out' / '001 OK Logout completed.'
+        // message when connecting via IMAP/143, but simply closes the connection via
+        // IMAPS/993.
         if (! socket_readln(socket, buf, sizeof(buf))) {
-                socket_setError(socket, "IMAP: logout response read error -- %s", STRERROR);
-                return FALSE;
-        }
+                /*
+                char *error = STRERROR;
+                if (! strncmp(error, success, strlen(success))) {
+                        socket_setError(socket, "IMAP: logout response read error -- %s", error);
+                        return FALSE;
+                } else {
+                        socket_setError(socket, "IMAP: logout response connection closed -- %s", error);
+                        return TRUE;
+                }
+                 */
+                socket_setError(socket, "IMAP: logout response connection closed -- %s", STRERROR);
+                return TRUE;
+        } else {
         Str_chomp(buf);
         if (strncasecmp(buf, bye, strlen(bye)) != 0) {
                 socket_setError(socket, "IMAP: invalid logout response: %s", buf);
                 return FALSE;
         }
+        }
 
         return TRUE;
 }
