--- monit-5.6/configure.ac.dist	2014-03-12 00:36:12.590403996 +0000
+++ monit-5.6/configure.ac	2014-03-12 00:37:58.224025072 +0000
@@ -771,10 +771,26 @@
                                    $maindir/lib/openssl \
                                    $maindir/lib/ssl \
                                    $maindir/ssl/lib \
+                                   $maindir/lib/32 \
+                                   $maindir/lib/32/openssl \
+                                   $maindir/lib/32/ssl \
+                                   $maindir/ssl/lib/32 \
+                                   $maindir/lib/x32 \
+                                   $maindir/lib/x32/openssl \
+                                   $maindir/lib/x32/ssl \
+                                   $maindir/ssl/lib/x32 \
                                    $maindir/lib/64 \
                                    $maindir/lib/64/openssl \
                                    $maindir/lib/64/ssl \
                                    $maindir/ssl/lib/64 \
+                                   $maindir/lib32 \
+                                   $maindir/lib32/openssl \
+                                   $maindir/lib32/ssl \
+                                   $maindir/ssl/lib32 \
+                                   $maindir/libx32 \
+                                   $maindir/libx32/openssl \
+                                   $maindir/libx32/ssl \
+                                   $maindir/ssl/libx32 \
                                    $maindir/lib64 \
                                    $maindir/lib64/openssl \
                                    $maindir/lib64/ssl \
