--- a/packaging/cmake/Modules/NetdataLibBPF.cmake
+++ b/packaging/cmake/Modules/NetdataLibBPF.cmake
@@ -66,20 +66,6 @@ function(netdata_bundle_libbpf)
     endif()
 
     set(_libbpf_library "${libbpf_SOURCE_DIR}/usr/${_libbpf_lib_dir}/libbpf.a")
-
-    ExternalProject_Add(
-        libbpf
-        GIT_REPOSITORY https://github.com/netdata/libbpf.git
-        GIT_TAG ${_libbpf_tag}
-        SOURCE_DIR "${libbpf_SOURCE_DIR}"
-        CONFIGURE_COMMAND mkdir -pv src/build/
-        BUILD_COMMAND ${MAKE_COMMAND} -C src CC=${CMAKE_C_COMPILER} BUILD_STATIC_ONLY=1 OBJDIR=build/ all
-        BUILD_IN_SOURCE 1
-        BUILD_BYPRODUCTS "${_libbpf_library}"
-        INSTALL_COMMAND ${MAKE_COMMAND} -C src CC=${CMAKE_C_COMPILER} BUILD_STATIC_ONLY=1 OBJDIR=build/ DESTDIR=../ install
-        EXCLUDE_FROM_ALL 1
-    )
-
     add_library(libbpf_library STATIC IMPORTED GLOBAL)
     set_property(
         TARGET libbpf_library
@@ -98,5 +84,5 @@ function(netdata_add_libbpf_to_target _t
     target_link_libraries(${_target} PUBLIC libbpf_library)
     target_include_directories(${_target} BEFORE PUBLIC "${NETDATA_LIBBPF_INCLUDE_DIRECTORIES}")
     target_compile_options(${_target} PUBLIC "${NETDATA_LIBBPF_COMPILE_OPTIONS}")
-    add_dependencies(${_target} libbpf)
+    add_dependencies(${_target} libbpf_library)
 endfunction()
