https://sourceforge.net/p/digimend/bugs/41/

--- a/usbhid-dump/lib/iface.c
+++ b/usbhid-dump/lib/iface.c
@@ -30,6 +30,7 @@
 #include <assert.h>
 #include <stdlib.h>
 #include <stdio.h>
+#include <stdint.h>
 #include <string.h>
 
 bool
--- a/usbhid-dump/lib/iface_list.c
+++ b/usbhid-dump/lib/iface_list.c
@@ -30,6 +30,7 @@
 #include <assert.h>
 #include <stdlib.h>
 #include <stdio.h>
+#include <stdint.h>
 
 bool
 uhd_iface_list_valid(const uhd_iface *list)
