diff --git a/debian.rockchip/config/config.common.ubuntu b/debian.rockchip/config/config.common.ubuntu
index 2173690e9a751..264578bc87fb9 100644
--- a/debian.rockchip/config/config.common.ubuntu
+++ b/debian.rockchip/config/config.common.ubuntu
@@ -125,7 +125,7 @@ CONFIG_RCU_NEED_SEGCBLIST=y
 
 CONFIG_IKCONFIG=y
 CONFIG_IKCONFIG_PROC=y
-# CONFIG_IKHEADERS is not set
+CONFIG_IKHEADERS=y
 CONFIG_LOG_BUF_SHIFT=18
 CONFIG_LOG_CPU_MAX_BUF_SHIFT=12
 CONFIG_PRINTK_SAFE_LOG_BUF_SHIFT=13
@@ -1617,7 +1617,7 @@ CONFIG_NET_ACT_SKBEDIT=m
 CONFIG_NET_ACT_CSUM=m
 # CONFIG_NET_ACT_MPLS is not set
 # CONFIG_NET_ACT_VLAN is not set
-# CONFIG_NET_ACT_BPF is not set
+CONFIG_NET_ACT_BPF=m
 # CONFIG_NET_ACT_CONNMARK is not set
 # CONFIG_NET_ACT_CTINFO is not set
 # CONFIG_NET_ACT_SKBMOD is not set
@@ -8632,7 +8632,13 @@ CONFIG_DEBUG_BUGVERBOSE=y
 #
 # Compile-time checks and compiler options
 #
-# CONFIG_DEBUG_INFO is not set
+CONFIG_DEBUG_INFO=y
+# CONFIG_DEBUG_INFO_REDUCED is not set
+# CONFIG_DEBUG_INFO_COMPRESSED is not set
+# CONFIG_DEBUG_INFO_SPLIT is not set
+# CONFIG_DEBUG_INFO_DWARF4 is not set
+CONFIG_DEBUG_INFO_BTF=y
+CONFIG_GDB_SCRIPTS=y
 CONFIG_ENABLE_MUST_CHECK=y
 CONFIG_FRAME_WARN=2048
 # CONFIG_STRIP_ASM_SYMS is not set
@@ -8845,6 +8851,7 @@ CONFIG_STRICT_DEVMEM=y
 # arm64 Debugging
 #
 # CONFIG_PID_IN_CONTEXTIDR is not set
+CONFIG_DEBUG_EFI=y
 # CONFIG_ARM64_RELOC_TEST is not set
 # CONFIG_CORESIGHT is not set
 # end of arm64 Debugging
diff --git a/debian.rockchip/rules.d/arm64.mk b/debian.rockchip/rules.d/arm64.mk
index 509bfa4438c5c..a9736560ed935 100644
--- a/debian.rockchip/rules.d/arm64.mk
+++ b/debian.rockchip/rules.d/arm64.mk
@@ -14,7 +14,7 @@ do_linux_tools		= true
 do_tools_usbip		= true
 do_tools_cpupower	= true
 do_tools_perf		= true
-do_tools_bpftool	= false
+do_tools_bpftool	= true
 
 do_common_headers_indep	= false
 
