diff --git a/debian.rockchip/abi/abiname b/debian.rockchip/abi/abiname
index 021ea30c0eb21..9f1d48efca0ec 100644
--- a/debian.rockchip/abi/abiname
+++ b/debian.rockchip/abi/abiname
@@ -1,2 +1,2 @@
-10
+11
 
diff --git a/debian.rockchip/abi/version b/debian.rockchip/abi/version
index 0a7015d3359fb..598fcea3e9256 100644
--- a/debian.rockchip/abi/version
+++ b/debian.rockchip/abi/version
@@ -1,2 +1,2 @@
-5.10.0-1010.10
+5.10.0-1011.11
 
diff --git a/debian.rockchip/changelog b/debian.rockchip/changelog
index 9d1f275b15508..dd2b3e45fe6cf 100644
--- a/debian.rockchip/changelog
+++ b/debian.rockchip/changelog
@@ -1,3 +1,12 @@
+linux-rockchip (5.10.0-1011.11) jammy; urgency=medium
+
+  * arm64: dts: rockchip: Add support for the ArmSoM AIM7
+  * Add DTS files for Firefly AIO-3588L
+  * arm64: dts: rockchip: add the Orange Pi CM5
+  * arm64: dts: rockchip: add the Orange Pi 5 Max
+
+ -- Joshua Riek <jjriek@verizon.net>  Sun, 28 Jul 2024 22:34:36 -0400
+
 linux-rockchip (5.10.0-1010.10) jammy; urgency=medium
 
   * phy: rockchip-samsung-hdptx-hdmi: get phy init status before register clk
