--- www/api.php	2014-07-29 02:14:47.000000000 +0100
+++ www/api.php	2014-07-29 21:30:01.649857515 +0100
@@ -29,7 +29,9 @@ You should have received a copy of the G
 along with NabaztagLives.  If not, see <http://www.gnu.org/licenses/>.
 */
 //error_reporting(E_ALL);  //don't use this in production
-ini_set('display_errors', 0);
+//ini_set('display_errors', 0);
+error_reporting(E_ALL);  //don't use this in production
+ini_set('display_errors', 1);
 ini_set('log_errors', 1);
 ini_set('error_log','logs/error.log');
 //error_reporting(0);  //don't use this, use ini_set or nothing goes to the log
@@ -57,7 +59,8 @@ if(isset($_GET['color'])) $color= $_GET[
 if(isset($_GET['play'])) $play=$_GET['play'];  //play mp3
 if(isset($_GET['stream'])) $stream=$_GET['stream']; //stream mp3
 
-$debug=false;
+//$debug=false;
+$debug=true;
 
 if(strlen($sn) == 0)
 {
@@ -104,7 +107,7 @@ include 'config/db.php';
 $con = mysqli_connect($host,$user,$pass,$db);
 if (!$con) 
 {
-	logError('Api.php: Could not connect: ' . mysqli_connect_errno() . mysqli_connect_error());
+	logError('api.php: Could not connect: ' . mysqli_connect_errno() . mysqli_connect_error());
 	echo 'The operation failed.  The head rabbit has been placed on notice.';
 	return;
 }
@@ -124,7 +127,7 @@ if($debug) echo $cmd . '<p>';
 $result = mysqli_query($con,$cmd);
 if (!$result) 
 {
-	logError('Api.php: Invalid log connect call. ' . mysqli_error($con));
+	logError('api.php: Invalid log connect call. ' . mysqli_error($con));
 	echo 'The operation failed.  The head rabbit has been placed on notice.';
 	return;
 }
@@ -132,7 +135,7 @@ if (!$result)
 $result = mysqli_query($con,"select @rabbitID");
 if (!$result) 
 {
-	logError('Api.php: Invalid log query. ' . mysqli_error($con));
+	logError('api.php: Invalid log query. ' . mysqli_error($con));
 	echo 'The operation failed.  The head rabbit has been placed on notice.';
 	return;
 }
@@ -148,7 +151,8 @@ if($rabbitID < 1)  //rows found in DB
 	return;
 }
 
-mysqli_next_result($con);  //required to avoid sync error
+//mysqli_next_result($con);  //required to avoid sync error
+while(mysqli_next_result($con));
 
 /*************************************
  * get rabbit
@@ -158,7 +162,7 @@ if($debug) echo $cmd . '<p>';
 $result = mysqli_query($con,$cmd);
 if (!$result) 
 {
-	logError('Api.php: Invalid getRabbit query. ' . mysqli_error($con));
+	logError('api.php: Invalid getRabbit query. ' . mysqli_error($con));
 	echo 'The operation failed.  The head rabbit has been placed on notice.';
 	return;
 }
@@ -187,11 +191,12 @@ while($row = mysqli_fetch_row($result))
 
 }
 
-mysqli_next_result($con);  //required to avoid sync error
+//mysqli_next_result($con);  //required to avoid sync error
+while(mysqli_next_result($con));
 
+/*
 if(($sn == "00904B8C6C6D") || ($sn == '000000000000'))
 {
-	/*
 	$out = 	'./vl/FR/bin2/api.bin';
 	$msg = file_get_contents($out);
 	//echo $msg . '<P>';
@@ -231,8 +236,8 @@ if(($sn == "00904B8C6C6D") || ($sn == '0
 	echo 'OK';
   mysqli_close($con);
 	exit(0);
-	*/
 }
+*/
 
 $min  = date("i"); //minute 00-59
 $sec  = date("s"); //secs 00-59
@@ -252,10 +257,11 @@ $cmd = "delete from queue "
 $result = mysqli_query($con,$cmd);
 if (!$result) 
 {
-	logError('Api.php: Invalid delete queue query: ' . mysqli_error($con));
+	logError('api.php: Invalid delete queue query: ' . mysqli_error($con));
 }
 
-mysqli_next_result($con);  //required to avoid sync error
+//mysqli_next_result($con);  //required to avoid sync error
+while(mysqli_next_result($con));
 
 //need to check here to make sure there is not already a msg in the queue from multiple attempts
 
@@ -283,7 +289,8 @@ if($count > 0)
 	exit(0);
 }
 	
-mysqli_next_result($con);  //required to avoid sync error
+//mysqli_next_result($con);  //required to avoid sync error
+while(mysqli_next_result($con));
 
 /**************************************************************
  * LED color
@@ -467,25 +474,26 @@ return;
 function queueV1($msg,$rabbitID,$min,$con)
 {
 	if(strlen($msg) < 1) return;
-	
-	mysqli_next_result($con);  //required to avoid sync error
-	
-	$cmd = "call sp_Queue2('" . $rabbitID . "'
-			              ,'" . $min . "'
-						  ,'" . $msg . "'
-						  ,@msg
-						  )";
+
+	logError("DEBUG: api.php:queueV1 started with msg '" . $msg . "'");
+
+	//mysqli_next_result($con);  //required to avoid sync error
+	while(mysqli_next_result($con));
+
+	$cmd = "call sp_Queue2('" . $rabbitID . "', '" . $min . "', '" . $msg . "', @msg)";
                              
-  	$result = queryWithRetry($con,$cmd,$rabbitID,"api.php queue function.");
-	
+	$result = queryWithRetry($con,$cmd,$rabbitID,"api.php queue function.");
+
 	if (!$result) 
 	{
-	  	logError("Api.php: Queue query failed. " . mysqli_error($con));
-		mysqli_next_result($con);  //required to avoid sync error
+		logError("api.php: Queue query failed: " . mysqli_error($con));
+		//mysqli_next_result($con);  //required to avoid sync error
+		while(mysqli_next_result($con));
 		return;
 	}
-	
-	mysqli_next_result($con);  //required to avoid sync error
+
+	//mysqli_next_result($con);  //required to avoid sync error
+	while(mysqli_next_result($con));
 
 	$result = mysqli_query($con,"select @msg");
 
@@ -493,10 +501,10 @@ function queueV1($msg,$rabbitID,$min,$co
 		$msg = $row[0];
 
 	if($msg != 'OK')
-		logError("Api.php: Queue function: $msg");
-	
-	mysqli_next_result($con);  //required to avoid sync error
-	
+		logError("api.php: Queue function: $msg");
+
+	//mysqli_next_result($con);  //required to avoid sync error
+	while(mysqli_next_result($con));
 }
 
 /*****************************************************************
@@ -530,11 +538,13 @@ function queueCmd($serNbr,$min,$msg,$con
 	if (!$result) 
 	{
 		logError('API.php: Queue function invalid insert after retry: ' . mysqli_error($con));
-		mysqli_next_result($con);  //required to avoid sync error
+		//mysqli_next_result($con);  //required to avoid sync error
+		while(mysqli_next_result($con));
 		return;
 	}
 	
-	mysqli_next_result($con);  //required to avoid sync error
+	//mysqli_next_result($con);  //required to avoid sync error
+	while(mysqli_next_result($con));
 
 	$result = mysqli_query($con,"select @msg");
 
@@ -548,7 +558,8 @@ function queueCmd($serNbr,$min,$msg,$con
 		logError($msg);
 	}
 	
-	mysqli_next_result($con);  //required to avoid sync error
+	//mysqli_next_result($con);  //required to avoid sync error
+	while(mysqli_next_result($con));
 
 }
 
@@ -562,41 +573,41 @@ function queue($serNbr,$min,$tts,$con,$l
 		echo "You need to enter some text to send to the rabbit. $back";
 		return;
 	}
-	
+
+	logError("DEBUG: api.php:queue started for serial '" . $serNbr . "', text '" . $tts . "'");
+
 	$tts = mysqli_real_escape_string($con, $tts);
 
 	$lang = getLanguage($lang);
-	
+
 	doTTS($tts,$lang,$serNbr);
-		
+
 	$hutch = "./hutch/$serNbr";
-    $msg="PLAY $hutch/tts.mp3";
+	$msg="PLAY $hutch/tts.mp3";
 
-	$cmd = "call sp_Queue('" . $serNbr . "'
-											 ,'" . $min . "'
-											 ,'" . $msg . "'
-											 ,@msg
-											 )";
+	$cmd = "call sp_Queue('" . $serNbr . "', '" . $min . "', '" . $msg . "', @msg)";
                                             
 	$result = mysqli_query($con,$cmd);
-	
+
 	//retry transaction
-	if(!$result)
+	if (!$result)
 	{
-	usleep(1000000); //1 sec
-	//usleep(100000);  //100 ms
-	
-	$result = mysqli_query($con,$cmd);
+		usleep(1000000); //1 sec
+		//usleep(100000);  //100 ms
+
+		$result = mysqli_query($con,$cmd);
 	}
-	
+
 	if (!$result) 
 	{
-		logError('API.php Queue function invalid insert after retry: ' . mysqli_error($con));
-		mysqli_next_result($con);  //required to avoid sync error
+		logError('api.php Queue function invalid insert after retry: ' . mysqli_error($con));
+		//mysqli_next_result($con);  //required to avoid sync error
+		while(mysqli_next_result($con));
 		return;
 	}
-	
-	mysqli_next_result($con);  //required to avoid sync error
+
+	//mysqli_next_result($con);  //required to avoid sync error
+	while(mysqli_next_result($con));
 
 	$result = mysqli_query($con,"select @msg");
 
@@ -609,9 +620,9 @@ function queue($serNbr,$min,$tts,$con,$l
 	{
 		logError($msg);
 	}
-	
-	mysqli_next_result($con);  //required to avoid sync error
 
+	//mysqli_next_result($con);  //required to avoid sync error
+	while(mysqli_next_result($con));
 }
 
 /***********************************************************
@@ -630,11 +641,13 @@ function outV1($out,$rabbitID,$min,$con)
 	if (!$result) 
 	{
 		logError('Out function invalid update last Command query: ' . mysqli_error($con));
-		mysqli_next_result($con);  //required to avoid sync error
+		//mysqli_next_result($con);  //required to avoid sync error
+		while(mysqli_next_result($con));
 		return;
 	}
 	
-	mysqli_next_result($con);  //required to avoid sync error
+	//mysqli_next_result($con);  //required to avoid sync error
+	while(mysqli_next_result($con));
 
 	$result = mysqli_query($con,"select @msg");
 
@@ -644,7 +657,8 @@ function outV1($out,$rabbitID,$min,$con)
 	if($msg != 'OK')
 		logError($msg);
 	
-	mysqli_next_result($con);  //required to avoid sync error
+	//mysqli_next_result($con);  //required to avoid sync error
+	while(mysqli_next_result($con));
 }
 
 
@@ -670,4 +684,4 @@ function doTTS($t,$lang,$sn)
 }	
 
 ?>
-</html>
\ No newline at end of file
+</html>
