Description: fix a spelling mistake
Origin: vendor
Bug-Debian: http://bugs.debian.org/607647
Forwarded: no
Author: Jakub Wilk <jwilk@debian.org>
Reviewed-by: gregor herrmann <gregoa@debian.org>
Last-Update: 2011-01-15

--- a/lib/IO/Socket/Multicast.pm
+++ b/lib/IO/Socket/Multicast.pm
@@ -88,7 +88,7 @@ sub get_if_addr {
   return $interface if length $interface == 16;
   croak "IO::Interface module not available; use IP addr for interface"
     unless $sock->can('if_addr');
-  croak "unknown or unconfigured interace $interface"
+  croak "unknown or unconfigured interface $interface"
     unless my $addr = $sock->if_addr($interface);
   croak "interface is not multicast capable"
     unless $interface eq 'any' or ($sock->if_flags($interface) & IFF_MULTICAST());
