--- lib/IO/Socket/Multicast.pm
+++ lib/IO/Socket/Multicast.pm
@@ -120,7 +120,7 @@ sub mcast_dest {
 
 sub mcast_send {
   my $sock = shift;
-  my $data = shift || croak 'usage: $sock->mcast_send($data [,$address])';
+  my $data = shift; # || croak 'usage: $sock->mcast_send($data [,$address])'; # Allow packets with a zero-byte payload
   $sock->mcast_dest(shift) if @_;
   my $dest = $sock->mcast_dest || croak "no destination specified with mcast_send() or mcast_dest()";
   return send($sock,$data,0,$dest);
