--- /var/tmp/portage/dev-perl/IO-Socket-Multicast-1.12/work/IO-Socket-Multicast-1.12/lib/IO/Socket/Multicast.pm.dist	2014-10-04 22:35:49.273194665 +0100
+++ /var/tmp/portage/dev-perl/IO-Socket-Multicast-1.12/work/IO-Socket-Multicast-1.12/lib/IO/Socket/Multicast.pm	2014-10-04 22:36:40.364266647 +0100
@@ -120,7 +120,7 @@ sub mcast_dest {
 
 sub mcast_send {
   my $sock = shift;
-  my $data = shift || croak 'usage: $sock->mcast_send($data [,$address])';
+  my $data = shift # || croak 'usage: $sock->mcast_send($data [,$address])'; # Allow packets with a zero-byte payload
   $sock->mcast_dest(shift) if @_;
   my $dest = $sock->mcast_dest || croak "no destination specified with mcast_send() or mcast_dest()";
   return send($sock,$data,0,$dest);
