https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=699827

--- ebtables-2.0.10.4/libebtc.c 2011-12-15 21:02:47.000000000 +0100
+++ ebtables-2.0.10.4-patch/libebtc.c   2013-02-05 17:44:04.000000000 +0100
@@ -1102,7 +1102,7 @@
 			/* check if we've dealt with this chain already */
 			if (entries2->hook_mask & (1<<i))
 				goto letscontinue;
-			entries2->hook_mask |= entries->hook_mask;
+			entries2->hook_mask |= entries->hook_mask & ~(1 << NF_BR_NUMHOOKS);
 			/* Jump to the chain, make sure we know how to get back */
 			stack[sp].chain_nr = chain_nr;
 			stack[sp].n = j;
