--- extensions/ebt_standard.c.dist	2015-03-11 22:23:10.623576752 +0000
+++ extensions/ebt_standard.c	2015-03-11 22:23:12.300447096 +0000
@@ -48,7 +48,7 @@ static void print(const struct ebt_u_ent
 		struct ebt_u_entries *entries;
 
 		entries = entry->replace->chains[verdict + NF_BR_NUMHOOKS];
-		printf("%s", entries->name);
+		printf("%s ", entries->name);
 		return;
 	}
 	if (verdict == EBT_CONTINUE)
