--- readline-6.2/rltty.c
+++ readline-6.2/rltty.c
@@ -37,9 +37,9 @@
 
 #include "rldefs.h"
 
-#if defined (GWINSZ_IN_SYS_IOCTL)
+#if defined (GWINSZ_IN_SYS_IOCTL) || ( defined (__APPLE__) && defined (__MACH__) )
 #  include <sys/ioctl.h>
-#endif /* GWINSZ_IN_SYS_IOCTL */
+#endif /* GWINSZ_IN_SYS_IOCTL || __APPLE__ && __MACH__ */
 
 #include "rltty.h"
 #include "readline.h"
--- readline-6.2/terminal.c
+++ readline-6.2/terminal.c
@@ -51,9 +51,9 @@
 /* System-specific feature definitions and include files. */
 #include "rldefs.h"
 
-#if defined (GWINSZ_IN_SYS_IOCTL) && !defined (TIOCGWINSZ)
+#if ( defined (GWINSZ_IN_SYS_IOCTL) && !defined (TIOCGWINSZ) ) || ( defined (__APPLE__) && defined (__MACH__) )
 #  include <sys/ioctl.h>
-#endif /* GWINSZ_IN_SYS_IOCTL && !TIOCGWINSZ */
+#endif /* GWINSZ_IN_SYS_IOCTL && !TIOCGWINSZ || __APPLE__ && __MACH__ */
 
 #include "rltty.h"
 #include "tcap.h"
