--- natpmp.c.dist	2013-11-23 00:29:25.086443778 +0000
+++ natpmp.c	2013-11-23 00:35:23.676921403 +0000
@@ -149,8 +149,8 @@ void ProcessIncomingNATPMPPacket(int s)
 	syslog(LOG_INFO, "NAT-PMP request received from %s:%hu %dbytes",
            senderaddrstr, ntohs(senderaddr.sin_port), n);
 	if(n<2 || ((((req[1]-1)&~1)==0) && n<12)) {
-		syslog(LOG_WARNING, "discarding NAT-PMP request (too short) %dBytes",
-		       n);
+		syslog(LOG_WARNING, "discarding NAT-PMP request (%d bytes - too short) from %s",
+		       n,senderaddrstr);
 		return;
 	}
 	if(req[1] & 128) {
@@ -166,8 +166,8 @@ void ProcessIncomingNATPMPPacket(int s)
 	*((uint32_t *)(resp+4)) = htonl(time(NULL) - startup_time);
 	if(req[0] > 0) {
 		/* invalid version */
-		syslog(LOG_WARNING, "unsupported NAT-PMP version : %u",
-		       (unsigned)req[0]);
+		syslog(LOG_WARNING, "unsupported NAT-PMP version '%u' (should be '0') from %s",
+		       (unsigned)req[0],senderaddrstr);
 		resp[3] = 1;	/* unsupported version */
 	} else switch(req[1]) {
 	case 0:	/* Public address request */
