--- a/glib/glib/gnulib/vasnprintf.c
+++ b/glib/glib/gnulib/vasnprintf.c
@@ -696,9 +696,13 @@ vasnprintf (char *resultbuf, size_t *len
 		  }
 		*p = dp->conversion;
 #if HAVE_SNPRINTF
+# if !defined(__APPLE__) && !(((__GLIBC__ > 2 || (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 3)) && !defined __UCLIBC__) || ((defined _WIN32 || defined __WIN32__) && ! defined __CYGWIN__))
 		p[1] = '%';
 		p[2] = 'n';
 		p[3] = '\0';
+# else
+		p[1] = '\0';
+# endif
 #else
 		p[1] = '\0';
 #endif
