--- gc.h.dist	2013-07-10 15:30:33.008780064 +0100
+++ gc.h	2013-07-10 15:30:59.315523982 +0100
@@ -2,7 +2,7 @@
 #ifndef RUBY_GC_H
 #define RUBY_GC_H 1
 
-#if defined(__x86_64__) && defined(__GNUC__)
+#if defined(__x86_64__) && !defined(__ILP32__) && defined(__GNUC__)
 #define SET_MACHINE_STACK_END(p) __asm__ volatile ("movq\t%%rsp, %0" : "=r" (*(p)))
 #elif defined(__i386) && defined(__GNUC__)
 #define SET_MACHINE_STACK_END(p) __asm__ volatile ("movl\t%%esp, %0" : "=r" (*(p)))
