--- a/baselayout-2.1-r1.ebuild
+++ b/baselayout-2.1-r1.ebuild
@@ -106,10 +106,12 @@ multilib_layout() {
 				case ${CHOST} in
 				i?86*|x86_64*|powerpc*|sparc*|s390*)
 					if [ -d "${prefix}lib32" ] ; then
-						rm -f "${prefix}lib32"/.keep
-						if ! rmdir "${prefix}lib32" 2>/dev/null ; then
-							ewarn "You need to merge ${prefix}lib32 into ${prefix}lib"
-							die "non-empty dir found where there should be none: ${prefix}lib32"
+						if ! [ "${LIBDIR_x32}" = "lib32" ] ; then
+							rm -f "${prefix}lib32"/.keep
+							if ! rmdir "${prefix}lib32" 2>/dev/null ; then
+								ewarn "You need to merge ${prefix}lib32 into ${prefix}lib"
+								die "non-empty dir found where there should be none: ${prefix}lib32"
+							fi
 						fi
 					fi
 					;;
