--- a/src/usr/share/rpimonitor/web/js/rpimonitor.utils.js
+++ b/src/usr/share/rpimonitor/web/js/rpimonitor.utils.js
@@ -73,7 +73,13 @@ function KMG(value, initPre){
 }
 
 function Percent(value,total){
-  return (100*value/total).toFixed(2)+"%";
+  // For graphs and reports, it's better for the result of dividing by zero
+  // to be zero rather than NaN...
+  var percent = 0
+  if (total != 0 ) {
+    percent = 100*value/total
+  }
+  return percent.toFixed(2)+"%";
 }
 
 function ProgressBar(value, max, warning, danger){
