--- llvm-3.4/tools/clang/lib/Driver/Tools.cpp.dist	2014-03-23 02:04:32.000000000 +0000
+++ llvm-3.4/tools/clang/lib/Driver/Tools.cpp	2014-03-23 02:04:51.000000000 +0000
@@ -5007,7 +5007,7 @@ void darwin::Link::ConstructJob(Compilat
                 CmdArgs.push_back("-lcrt1.o");
               else if (getDarwinToolChain().isMacosxVersionLT(10, 6))
                 CmdArgs.push_back("-lcrt1.10.5.o");
-              else if (getDarwinToolChain().isMacosxVersionLT(10, 8))
+              else
                 CmdArgs.push_back("-lcrt1.10.6.o");
 
               // darwin_crt2 spec is empty.
