--- llvm-3.5.0.src/tools/clang/lib/Driver/ToolChains.cpp.dist	2014-09-18 23:59:04.000000000 +0100
+++ llvm-3.5.0.src/tools/clang/lib/Driver/ToolChains.cpp	2014-09-19 00:00:07.000000000 +0100
@@ -1090,7 +1090,8 @@ void Darwin::addStartObjectFileArgs(cons
               CmdArgs.push_back("-lcrt1.o");
             else if (isMacosxVersionLT(10, 6))
               CmdArgs.push_back("-lcrt1.10.5.o");
-            else if (isMacosxVersionLT(10, 8))
+            //else if (isMacosxVersionLT(10, 8))
+            else
               CmdArgs.push_back("-lcrt1.10.6.o");
 
             // darwin_crt2 spec is empty.
