--- a/tools/clang/lib/Frontend/InitHeaderSearch.cpp
+++ b/tools/clang/lib/Frontend/InitHeaderSearch.cpp
@@ -220,6 +220,7 @@ void InitHeaderSearch::AddDefaultCInclud
       if (triple.getEnvironment() != llvm::Triple::Cygnus)
         break;
     default:
+      AddPath("@GENTOO_PORTAGE_EPREFIX@/usr/include", System, false);
       // FIXME: temporary hack: hard-coded paths.
       AddPath("/usr/local/include", System, false);
       break;
@@ -469,6 +470,7 @@ void InitHeaderSearch::AddDefaultInclude
   // Add the default framework include paths on Darwin.
   if (HSOpts.UseStandardSystemIncludes) {
     if (triple.isOSDarwin()) {
+      AddPath("@GENTOO_PORTAGE_EPREFIX@/Frameworks", System, true);
       AddPath("/System/Library/Frameworks", System, true);
       AddPath("/Library/Frameworks", System, true);
     }
