--- gdbm-1.11/src/gdbmtool.h
+++ gdbm-1.11/src/gdbmtool.h
@@ -237,6 +237,7 @@ extern struct dsegm *dsdef[];
 #define VAR_ERR_BADVALUE 4       /* Only for variable_set: the value is
 				    not valid for this variable. */
 
+int variable_unset (const char *name);
 int variable_set (const char *name, int type, void *val);
 int variable_get (const char *name, int type, void **val);
 int variable_is_set (const char *name);
