--- master/lib/Munin/Master/HTMLOld.pm.dist	2014-07-14 20:33:24.687042507 +0100
+++ master/lib/Munin/Master/HTMLOld.pm	2014-07-14 20:44:43.033773890 +0100
@@ -710,7 +710,10 @@ sub emit_main_index {
 
 
 sub copy_web_resources {
-    my ($staticdir, $htmldir) = @_;
+	my ($staticdir, $htmldir) = @_;
+
+	return( undef ) if( $staticdir =~ m|$htmldir/+static/*$| );
+
 	unless(dircopy($staticdir, "$htmldir/static")){
 		ERROR "[ERROR] Could not copy contents from $staticdir to $htmldir";
 		die "[ERROR] Could not copy contents from $staticdir to $htmldir";
