--- master/lib/Munin/Master/Config.pm.dist	2014-07-14 21:12:31.418081795 +0100
+++ master/lib/Munin/Master/Config.pm	2014-07-14 21:29:36.042865350 +0100
@@ -436,7 +436,7 @@ sub _split_config_line {
 sub _split_config_line_ok {
     # Split config line and do some extra syntaxy checks
     #
-    # If all is not well we'll corak here.
+    # If all is not well we'll croak here.
 
     my ($self,$longkey,$value) = @_;
    
@@ -460,6 +460,10 @@ sub _split_config_line_ok {
 	ERROR "[ERROR] Hostname '$host' contains illegal characters (http://en.wikipedia.org/wiki/Hostname#Restrictions_on_valid_host_names).  Please fix this by replacing illegal characters with '-'.  Remember to do it on both in the master configuration and on the munin-node.";
 	croak "[ERROR] Hostname '$host' contains illegal characters (http://en.wikipedia.org/wiki/Hostname#Restrictions_on_valid_host_names).  Please fix this by replacing illegal characters with '-'.  Remember to do it on both in the master configuration and on the munin-node.\n";
     }
+    if ($host =~ m/^(?:config|static|templates)$/) {
+	ERROR "[ERROR] Hostname '$host' matches reserved word 'config', 'static', or 'template'.  Please rename or omit these hosts.  Remember to do this both in the master configuration and on the munin-node.";
+	croak "[ERROR] Hostname '$host' matches reserved word 'config', 'static', or 'template'.  Please rename or omit these hosts.  Remember to do this both in the master configuration and on the munin-node.\n";
+    }
 
     return ($groups,$host,$key);
 }
