--- ./opiegen.c.dist	2013-01-23 00:06:37.983308379 +0000
+++ ./opiegen.c	2013-01-23 00:10:36.962429442 +0000
@@ -18,6 +18,12 @@
 */
 #include "opie_cfg.h"
 #include <stdio.h>
+#if HAVE_STDLIB_H
+#include <stdlib.h>
+#endif /* HAVE_STDLIB_H */
+#if HAVE_STRING_H
+#include <string.h>
+#endif /* HAVE_STRING_H */
 #if DEBUG
 #include <syslog.h>
 #endif /* DEBUG */
--- ./libopie/lookup.c.dist	2013-01-23 00:12:18.377107442 +0000
+++ ./libopie/lookup.c	2013-01-23 00:12:48.792389436 +0000
@@ -13,6 +13,9 @@
 
 #include "opie_cfg.h"
 #include <stdio.h>
+#if HAVE_STRING_H
+#include <string.h>
+#endif /* HAVE_STRING_H */
 #include "opie.h"
 
 int opielookup FUNCTION((opie, principal), struct opie *opie AND char *principal)
--- ./libopie/newseed.c.dist	2013-01-23 00:13:51.496201014 +0000
+++ ./libopie/newseed.c	2013-01-23 00:14:25.164668138 +0000
@@ -14,6 +14,7 @@
 */
 
 #include "opie_cfg.h"
+#include <stdio.h>
 #if HAVE_TIME_H
 #include <time.h>
 #endif /* HAVE_TIME_H */
--- ./libopie/randomchallenge.c.dist	2013-01-23 00:15:19.722184072 +0000
+++ ./libopie/randomchallenge.c	2013-01-23 00:19:25.964305486 +0000
@@ -26,6 +26,10 @@
 */
 
 #include "opie_cfg.h"
+#include <stdio.h>
+#if HAVE_STRING_H
+#include <string.h>
+#endif /* HAVE_STRING_H */
 #include "opie.h"
 
 static char *algids[] = { 0, 0, 0, "sha1", "md4", "md5" };
--- ./libopie/generator.c.dist	2013-01-23 00:15:54.000623296 +0000
+++ ./libopie/generator.c	2013-01-23 00:20:58.960071571 +0000
@@ -28,6 +28,7 @@
 */
 
 #include "opie_cfg.h"
+#include <stdio.h>
 #if HAVE_STRING_H
 #include <string.h>
 #endif /* HAVE_STRING_H */
--- ./libopie/passwd.c.dist	2013-01-23 00:15:06.676111389 +0000
+++ ./libopie/passwd.c	2013-01-23 00:18:34.706639522 +0000
@@ -21,6 +21,9 @@
 */
 
 #include "opie_cfg.h"
+#if HAVE_STRING_H
+#include <string.h>
+#endif /* HAVE_STRING_H */
 #include "opie.h"
 
 int opiepasswd FUNCTION((old, flags, principal, n, seed, ks), struct opie *old AND int flags AND char *principal AND int n AND char *seed AND char *ks)
--- ./libopie/version.c.dist	2013-01-23 00:15:40.424574924 +0000
+++ ./libopie/version.c	2013-01-23 00:20:34.024540297 +0000
@@ -18,6 +18,10 @@
         Created at NRL for OPIE 2.2 from opiesubr.c.
 */
 #include "opie_cfg.h"
+#include <stdio.h>
+#if HAVE_STDLIB_H
+#include <stdlib.h>
+#endif /* HAVE_STDLIB_H */
 #include "opie.h"
 
 VOIDRET opieversion FUNCTION_NOARGS
--- ./opieinfo.c.dist	2013-01-23 00:08:00.732873746 +0000
+++ ./opieinfo.c	2013-01-23 00:11:03.387188607 +0000
@@ -33,6 +33,9 @@
 
 #include "opie_cfg.h"
 #include <stdio.h>
+#if HAVE_STDLIB_H
+#include <stdlib.h>
+#endif /* HAVE_STDLIB_H */
 #include <errno.h>
 #if HAVE_UNISTD_H
 #include <unistd.h>
--- ./opietest.c.dist	2013-01-23 00:06:00.141697911 +0000
+++ ./opietest.c	2013-01-23 00:11:19.096473296 +0000
@@ -27,6 +27,9 @@
 */
 #include "opie_cfg.h"
 #include <stdio.h>
+#if HAVE_STDLIB_H
+#include <stdlib.h>
+#endif /* HAVE_STDLIB_H */
 #include "opie.h"
 
 char buffer[1024];
