--- src/asn1-gentables.c.dist	2014-12-03 23:37:49.000000000 +0000
+++ src/asn1-gentables.c	2014-12-03 23:39:36.000000000 +0000
@@ -280,13 +280,21 @@ one_file (const char *fname, int *count,
   ksba_asn_tree_t tree;
   int rc;
 
+  #pragma clang diagnostic push
+  #pragma clang diagnostic ignored "-Wall"
+  #pragma clang diagnostic ignored "-Wimplicit-function-declaration"
   rc = ksba_asn_parse_file (fname, &tree, check_only);
+  #pragma clang diagnostic pop
   if (rc)
     print_error ("error parsing `%s': %s\n", fname, gpg_strerror (rc) );
   else if (!check_only)
     {
       if (dump_only)
+        #pragma clang diagnostic push
+        #pragma clang diagnostic ignored "-Wall"
+        #pragma clang diagnostic ignored "-Wimplicit-function-declaration"
         ksba_asn_tree_dump (tree, dump_only==2? "<":NULL, fp);
+        #pragma clang diagnostic pop
       else
         {
           if (!*count)
