--- rpimonitor/rpimonitord.dist	2015-03-14 19:01:48.049612000 +0000
+++ rpimonitor/rpimonitord	2015-03-14 19:04:23.919612000 +0000
@@ -61,8 +61,10 @@
   $_ = abs_path($0);
   my ($path,$file) = /(.*)\/([^\/]*)$/;
 
-  push(@{$this->{'daemon'}->{'confFiles'}},"/etc/rpimonitor/data.conf");
-  push(@{$this->{'daemon'}->{'confFiles'}},"/etc/rpimonitor/daemon.conf");
+  if( scalar( @{$this->{'daemon'}->{'confFiles'}} < 2 )) {
+    push(@{$this->{'daemon'}->{'confFiles'}},"/etc/rpimonitor/data.conf") if -r "/etc/rpimonitor/data.conf";
+    push(@{$this->{'daemon'}->{'confFiles'}},"/etc/rpimonitor/daemon.conf") if -r "/etc/rpimonitor/daemon.conf";
+  }
 
   foreach ( @{$this->{'daemon'}->{'confFiles'}} ) {
     #print "$_\n";
@@ -151,11 +151,11 @@ sub LoadFile
   $this->Debug(2,"Loading file: $confFile");
 
   open ( FILE, $confFile)
-    or die "Error while openning configuration file \"$confFile\" because $!\n";
+    or die "Error while opening configuration file \"$confFile\" because $!\n";
   while (<FILE>){
     chomp;
     /^\s*#|^$/ and next;                 # Ignore comments
-    /^include=(.*)/ and -f $1 and push($this->{'daemon'}->{'confFiles'}, $1) and next;
+    /^include=(.*)/ and -f $1 and push(@{$this->{'daemon'}->{'confFiles'}}, $1) and next;
     my ($key, $value) = /^([^=]*)=(.*)/; # Extract key and value
     my @leaves=split('\.', $key);        # Explode key and construct config hash
     my $tree=$this;
