diff --git a/drivers/pinctrl/cix/pinctrl-sky1-base.c b/drivers/pinctrl/cix/pinctrl-sky1-base.c
--- a/drivers/pinctrl/cix/pinctrl-sky1-base.c
+++ b/drivers/pinctrl/cix/pinctrl-sky1-base.c
@@ -376,7 +376,6 @@ static void sky1_pinconf_dbg_show(struct pinctrl_dev *pctldev,
 static void sky1_pinconf_group_dbg_show(struct pinctrl_dev *pctldev,
 					  struct seq_file *s, unsigned int group)
 {
-	struct sky1_pinctrl *spctl = pinctrl_dev_get_drvdata(pctldev);
 	struct group_desc *grp;
 	unsigned long config;
 	const char *name;
