diff --git a/drivers/acpi/processor_perflib.c b/drivers/acpi/processor_perflib.c
--- a/drivers/acpi/processor_perflib.c
+++ b/drivers/acpi/processor_perflib.c
@@ -43,8 +43,8 @@
  */
 static int ignore_ppc = -1;
 module_param(ignore_ppc, int, 0644);
-MODULE_PARM_DESC(ignore_ppc, "If the frequency of your machine gets wrongly" \
-		 "limited by BIOS, this should help");
+MODULE_PARM_DESC(ignore_ppc,
+		 "Set to 1 to ignore ACPI _PPC CPU performance limits");
 
 static bool acpi_processor_ppc_in_use;
 
