diff --git a/drivers/acpi/processor_thermal.c b/drivers/acpi/processor_thermal.c
index 3d5d20056..f63c0df32 100644
--- a/drivers/acpi/processor_thermal.c
+++ b/drivers/acpi/processor_thermal.c
@@ -65,7 +65,8 @@ static int phys_package_first_cpu(int cpu)
 #ifdef CONFIG_CIX_THERMAL
 static unsigned int cix_get_static_power(int cpu)
 {
-	int pcpu = MPIDR_AFFINITY_LEVEL(cpu_logical_map(cpu), 1);
+	//int pcpu = MPIDR_AFFINITY_LEVEL(cpu_logical_map(cpu), 1);
+	int pcpu = cpu;
 	unsigned long long temp;
 	struct acpi_object_list arg_list;
 	union acpi_object arg;
@@ -76,7 +77,8 @@ static unsigned int cix_get_static_power(int cpu)
 	arg_list.pointer = &arg;
 	arg.type = ACPI_TYPE_INTEGER;
 	arg.integer.value = (u64)pcpu;
-	if (pcpu > 12) {
+	//if (pcpu > 12) {
+	if (pcpu >= num_possible_cpus()) {
 		pr_err("CIX: Invalid CPU[%d %d] for SPRG\n", pcpu, cpu);
 		return 0;
 	}
@@ -93,7 +95,8 @@ static unsigned int cix_get_static_power(int cpu)
 
 static unsigned int cix_get_dynamic_power(int cpu)
 {
-	int pcpu = MPIDR_AFFINITY_LEVEL(cpu_logical_map(cpu), 1);
+	//int pcpu = MPIDR_AFFINITY_LEVEL(cpu_logical_map(cpu), 1);
+	int pcpu = cpu;
 	unsigned long long temp;
 	struct acpi_object_list arg_list;
 	union acpi_object arg;
@@ -104,7 +107,8 @@ static unsigned int cix_get_dynamic_power(int cpu)
 	arg_list.pointer = &arg;
 	arg.type = ACPI_TYPE_INTEGER;
 	arg.integer.value = (u64)pcpu;
-	if (pcpu > 12) {
+	//if (pcpu > 12) {
+	if (pcpu >= num_possible_cpus()) {
 		pr_err("CIX: Invalid CPU[%d %d] for DPRG\n", pcpu, cpu);
 		return 0;
 	}
@@ -372,6 +372,7 @@ processor_set_cur_state(struct thermal_cooling_device *cdev,
 static int processor_get_requested_power(struct thermal_cooling_device *cdev,
 					 u32 *power)
 {
+#if defined(CONFIG_CPU_FREQ) && defined(CONFIG_CIX_THERMAL)
 	struct acpi_device *device = cdev->devdata;
 	struct cpufreq_policy *policy;
 	struct acpi_processor *pr;
@@ -395,6 +396,9 @@ static int processor_get_requested_power(struct thermal_cooling_device *cdev,
 	cpufreq_cpu_put(policy);
 
 	return 0;
+#else
+	return -EINVAL;
+#endif
 }
 
 /**
@@ -413,6 +417,7 @@ static int processor_get_requested_power(struct thermal_cooling_device *cdev,
 static int processor_state2power(struct thermal_cooling_device *cdev,
 				 unsigned long state, u32 *power)
 {
+#if defined(CONFIG_CPU_FREQ) && defined(CONFIG_CIX_THERMAL)
 	unsigned int freq, opp_power, num_cpus, idx;
 	struct cpufreq_policy *policy = NULL;
 	struct cppc_cpudata *cpu_data;
@@ -466,6 +471,9 @@ static int processor_state2power(struct thermal_cooling_device *cdev,
 EXIT:
 	cpufreq_cpu_put(policy);
 	return ret;
+#else
+	return -EINVAL;
+#endif
 }
 
 /**
@@ -486,6 +494,7 @@ static int processor_state2power(struct thermal_cooling_device *cdev,
 static int processor_power2state(struct thermal_cooling_device *cdev,
 				 u32 power, unsigned long *state)
 {
+#if defined(CONFIG_CPU_FREQ) && defined(CONFIG_CIX_THERMAL)
 	struct cpufreq_policy *policy;
 	struct cppc_cpudata *cpu_data;
 	struct em_perf_domain *em;
@@ -543,6 +552,9 @@ static int processor_power2state(struct thermal_cooling_device *cdev,
 
 	*state = cpu_data->opp_level_num - i - 1;
 	return 0;
+#else
+	return -EINVAL;
+#endif
 }
 
 
@@ -620,3 +620,4 @@ void acpi_processor_thermal_exit(struct acpi_processor *pr,
 		pr->cdev = NULL;
 	}
 }
+EXPORT_SYMBOL_GPL(acpi_processor_thermal_exit);
