diff --git a/drivers/gpu/drm/panthor/panthor_fw.c b/drivers/gpu/drm/panthor/panthor_fw.c
index 18358da795e1..f525c5c8ad12 100644
--- a/drivers/gpu/drm/panthor/panthor_fw.c
+++ b/drivers/gpu/drm/panthor/panthor_fw.c
@@ -752,9 +752,14 @@ static int panthor_fw_load(struct panthor_device *ptdev)
 	const struct firmware *fw = NULL;
 	struct panthor_fw_binary_iter iter = {};
 	struct panthor_fw_binary_hdr hdr;
-	char fw_path[20] = "mali_csffw.bin";
+	char fw_path[128];
 	int ret;
 
+	snprintf(fw_path, sizeof(fw_path), "arm/mali/arch%d.%d/%s",
+		 (u32)GPU_ARCH_MAJOR(ptdev->gpu_info.gpu_id),
+		 (u32)GPU_ARCH_MINOR(ptdev->gpu_info.gpu_id),
+		 CSF_FW_NAME);
+
 	ret = request_firmware(&fw, fw_path, ptdev->base.dev);
 	if (ret) {
 		drm_err(&ptdev->base, "Failed to load firmware image '%s'\n",
@@ -1422,3 +1427,4 @@ int panthor_fw_init(struct panthor_device *ptdev)
 }
 
 MODULE_FIRMWARE("arm/mali/arch10.8/mali_csffw.bin");
+MODULE_FIRMWARE("arm/mali/arch12.8/mali_csffw.bin");
