diff --git a/drivers/misc/armchina-npu/aipu_mm.c b/drivers/misc/armchina-npu/aipu_mm.c
--- a/drivers/misc/armchina-npu/aipu_mm.c
+++ b/drivers/misc/armchina-npu/aipu_mm.c
@@ -266,7 +266,7 @@ int aipu_mm_hold_tcb_buf_alloc(struct aipu_memory_manager *mm, struct aipu_job *
 	return ret;
 }
 
-void aipu_mm_hold_tcb_buf_free(struct aipu_memory_manager *mm)
+static void aipu_mm_hold_tcb_buf_free(struct aipu_memory_manager *mm)
 {
 	struct aipu_hold_tcb_buf *htbuf_head = mm->hold_tcb_head;
 	struct aipu_hold_tcb_buf *htbuf = NULL;
@@ -1615,7 +1615,7 @@ int aipu_mm_free(struct aipu_memory_manager *mm, struct aipu_buf_desc *buf, stru
 	return ret;
 }
 
-void tensor_dcache_inval_poc(void *start, void *end) {
+static void tensor_dcache_inval_poc(void *start, void *end) {
     uintptr_t line_size, addr_start, addr_end, tmp;
 
     line_size = cache_line_size();
diff --git a/drivers/misc/armchina-npu/sky1/sky1.c b/drivers/misc/armchina-npu/sky1/sky1.c
--- a/drivers/misc/armchina-npu/sky1/sky1.c
+++ b/drivers/misc/armchina-npu/sky1/sky1.c
@@ -280,13 +280,8 @@ int sky1_npu_pm_runtime_get_sync(struct device *dev)
 int sky1_npu_pm_runtime_put(struct device *dev)
 {
 #ifdef CONFIG_PM
-	int ret = 0;
-
-	ret = pm_runtime_put(dev);
-	if (ret < 0)
-		dev_err(dev, "PM runtime put failed! ret=%d", ret);
-
-	return ret;
+	pm_runtime_put(dev);
+	return 0;
 #else /* !CONFIG_PM  */
 	return 0;
 #endif /* CONFIG_PM */
@@ -486,13 +481,8 @@ static int sky1_npu_runtime_suspend(struct device *dev)
 	}
 
 	if (has_acpi_companion(dev)) {
-		for (int i = 0; i < CIX_NPU_PD_NUM; i++) {
-			ret = pm_runtime_put(cix_aipu_priv->pd_core[i]);
-			if (ret < 0) {
-				dev_err(cix_aipu_priv->pd_core[i], "NPU core PM runtime put failed! ret=%d", ret);
-				return ret;
-			}
-		}
+		for (int i = 0; i < CIX_NPU_PD_NUM; i++)
+			pm_runtime_put(cix_aipu_priv->pd_core[i]);
 	}
 
 	return ret;
