--- a/drivers/pci/controller/cadence/pci-sky1.c
+++ b/drivers/pci/controller/cadence/pci-sky1.c
@@ -2060,6 +2060,7 @@ bool sky1_pcie_link_up(struct cdns_pcie
 	return val & LINK_COMPLETE;
 }
 
+#ifdef CONFIG_PM_SLEEP
 static void sky1_pcie_pme_turn_off(struct sky1_pcie *pcie)
 {
 	u32 offset;
@@ -2069,6 +2070,7 @@ static void sky1_pcie_pme_turn_off(struc
 	writel(0, pcie->msg_base + offset);
 	usleep_range(1000, 10000);
 }
+#endif
 
 static const struct cdns_pcie_ops sky1_pcie_ops = {
 	.start_link = sky1_pcie_start_link,
@@ -2094,6 +2096,7 @@ static void sky1_pcie_set_linksta_slc(st
 	sky1_pcie_ctrl_writel_reg(pcie, pcie->linkctrl_offset, val);
 }
 
+#ifdef CONFIG_PM_SLEEP
 static void sky1_pcie_set_link_disable(struct sky1_pcie *pcie)
 {
 	void __iomem *status_addr;
@@ -2111,6 +2114,7 @@ static void sky1_pcie_set_link_disable(s
 	if (ret)
 		dev_dbg(pcie->dev, "link not in DISABLED, 0x%x\n", val);
 }
+#endif
 
 void __iomem *sky1_pcie_own_conf_map_bus(struct pci_bus *bus,
 					 unsigned int devfn, int where)
