--- lib/md5.c.dist	2014-12-06 20:31:29.000000000 +0000
+++ lib/md5.c	2014-12-06 20:31:35.000000000 +0000
@@ -183,7 +183,7 @@ cvs_MD5Final (digest, ctx)
 	putu32(ctx->buf[1], digest + 4);
 	putu32(ctx->buf[2], digest + 8);
 	putu32(ctx->buf[3], digest + 12);
-	memset(ctx, 0, sizeof(ctx));	/* In case it's sensitive */
+	memset(ctx, 0, sizeof(*ctx));	/* In case it's sensitive */
 }
 
 #ifndef ASM_MD5
