Description: make count if braces more balanced
Origin: vendor
Bug: https://rt.cpan.org/Public/Bug/Display.html?id=97190
Forwarded: https://rt.cpan.org/Public/Bug/Display.html?id=97190
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=754691
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2014-07-23

--- a/Interface.xs
+++ b/Interface.xs
@@ -595,6 +595,7 @@
        }
        ifap = ifap -> ifa_next;
      } 
+     }
      freeifaddrs(ifap);
 
      s = hwaddr;
