--- Libc-997.1.1/darwin/_dirhelper.c.dist	2014-11-18 18:31:04.000000000 +0000
+++ Libc-997.1.1/darwin/_dirhelper.c	2014-11-18 18:31:06.000000000 +0000
@@ -42,7 +42,7 @@
 #include <TargetConditionals.h>
 #include <xpc/xpc.h>
 #include <xpc/private.h>
-#if !TARGET_OS_IPHONE
+#if 0 && !TARGET_OS_IPHONE
 #include <CrashReporterClient.h>
 #endif /* !TARGET_OS_IPHONE */
 
@@ -79,11 +79,12 @@ static char *userdir = NULL;
 // 32 characters.
 static const char encode[] = "0123456789_bcdfghjklmnpqrstvwxyz";
 
-#if TARGET_OS_IPHONE
+//#if TARGET_OS_IPHONE
 
 #define setcrashlogmessage(fmt, ...) /* nothing */
 
-#else /* !TARGET_OS_IPHONE */
+//#else /* !TARGET_OS_IPHONE */
+#if 0
 
 static void
 encode_uuid_uid(const uuid_t uuid, uid_t uid, char *str)
--- Libc-997.1.1/os/assumes.h.dist	2014-11-18 18:33:42.000000000 +0000
+++ Libc-997.1.1/os/assumes.h	2014-11-18 18:33:46.000000000 +0000
@@ -64,9 +64,10 @@ struct _os_redirect_assumes_s {
  */
 typedef bool (*os_log_callout_t)(_SIMPLE_STRING asl_message, void *ctx, const char *);
 
-#if TARGET_OS_IPHONE
+//#if TARGET_OS_IPHONE
 #define os_set_crash_message(arg) /* no-op */
-#else
+//#else
+#if 0
 #include <CrashReporterClient.h>
 #define os_set_crash_message(arg) _crc_make_setter(message, (arg))
 #endif
--- dyld-239.3/src/ImageLoader.h.dist	2014-11-18 18:28:34.000000000 +0000
+++ dyld-239.3/src/ImageLoader.h	2014-11-18 18:28:47.000000000 +0000
@@ -40,7 +40,7 @@
 #include <vector>
 #include <new>
 
-#if __MAC_OS_X_VERSION_MIN_REQUIRED
+#if 0 && __MAC_OS_X_VERSION_MIN_REQUIRED
 	#include <CrashReporterClient.h>
 #else
 	// work around until iOS has CrashReporterClient.h
--- ld64-136/src/ld/Options.cpp.dist	2014-11-18 18:22:27.000000000 +0000
+++ ld64-136/src/ld/Options.cpp	2014-11-18 18:26:06.000000000 +0000
@@ -54,7 +54,7 @@ namespace lto {
 // magic to place command line in crash reports
 const int crashreporterBufferSize = 2000;
 static char crashreporterBuffer[crashreporterBufferSize];
-#if __MAC_OS_X_VERSION_MIN_REQUIRED >= 1070
+#if 0 && __MAC_OS_X_VERSION_MIN_REQUIRED >= 1070
 	#include <CrashReporterClient.h>
 	// hack until ld does not need to build on 10.6 anymore
     struct crashreporter_annotations_t gCRAnnotations 
@@ -4263,7 +4263,7 @@ void Options::checkForClassic(int argc,
 	bool newLinker = false;
 	
 	// build command line buffer in case ld crashes
-#if __MAC_OS_X_VERSION_MIN_REQUIRED >= 1070
+#if 0 && __MAC_OS_X_VERSION_MIN_REQUIRED >= 1070
 	CRSetCrashLogMessage(crashreporterBuffer);
 #endif
 	const char* srcRoot = getenv("SRCROOT");
