diff -urN squirrelmail-imap_proxy-1.2.8.dist/src/imapcommon.c squirrelmail-imap_proxy-1.2.8/src/imapcommon.c
--- squirrelmail-imap_proxy-1.2.8.dist/src/imapcommon.c	2017-01-27 14:53:57.000000000 -0600
+++ squirrelmail-imap_proxy-1.2.8/src/imapcommon.c	2017-10-22 19:12:09.344808120 -0500
@@ -554,16 +554,18 @@
     unsigned int Expiration;
     struct addrinfo *useai;
 
-    EVP_MD_CTX mdctx;
-    int md_len;
+    EVP_MD_CTX *mdctx;
+    unsigned int md_len;
 
     Expiration = PC_Struct.cache_expiration_time;
     memset( &Server, 0, sizeof Server );
     
     /* need to md5 the passwd regardless, so do that now */
-    EVP_DigestInit(&mdctx, EVP_md5());
-    EVP_DigestUpdate(&mdctx, Password, strlen(Password));
-    EVP_DigestFinal(&mdctx, md5pw, &md_len);
+    mdctx = EVP_MD_CTX_create ();
+    EVP_DigestInit_ex(mdctx, EVP_md5(), NULL);
+    EVP_DigestUpdate(mdctx, Password, strlen(Password));
+    EVP_DigestFinal(mdctx, md5pw, &md_len);
+    EVP_MD_CTX_destroy (mdctx);
     
     /* see if we have a reusable connection available */
     ICC_Active = NULL;
diff -urN squirrelmail-imap_proxy-1.2.8.dist/src/main.c squirrelmail-imap_proxy-1.2.8/src/main.c
--- squirrelmail-imap_proxy-1.2.8.dist/src/main.c	2017-01-27 14:53:57.000000000 -0600
+++ squirrelmail-imap_proxy-1.2.8/src/main.c	2017-10-22 18:50:41.426510268 -0500
@@ -460,7 +460,7 @@
     /* Set up OpenSSL thread protection */
     ssl_thread_setup(fn);
 
-#ifndef HAVE_RAND_EGD
+#ifdef HAVE_RAND_EGD
     if ( RAND_egd( ( RAND_file_name( f_randfile, sizeof( f_randfile ) ) == f_randfile ) ? f_randfile : "/.rnd" ) ) 
 #endif
     {
@@ -1574,9 +1574,9 @@
 	    verify_error = X509_V_ERR_CERT_CHAIN_TOO_LONG;
 	}
     }
-    switch (ctx->error) {
+    switch (err) {
     case X509_V_ERR_UNABLE_TO_GET_ISSUER_CERT:
-	X509_NAME_oneline(X509_get_issuer_name(ctx->current_cert), buf, sizeof(buf));
+	X509_NAME_oneline(X509_get_issuer_name(err_cert), buf, sizeof(buf));
 	syslog(LOG_NOTICE, "issuer= %s", buf);
 	break;
     case X509_V_ERR_CERT_NOT_YET_VALID:
