--- a/src/request.c
+++ b/src/request.c
@@ -515,7 +516,7 @@
     if ( !Username )
     {
 	snprintf( SendBuf, BufLen, "\n\n-----> C= %d %s PROXY: user tracing disabled. Expect further output until client logout.\n", (int)time(0), TraceUser );
-	write( Tracefd, SendBuf, strlen( SendBuf ) );
+	(void) !write( Tracefd, SendBuf, strlen( SendBuf ) );
 	
 	memset( TraceUser, 0, sizeof TraceUser );
 	snprintf( SendBuf, BufLen, "%s OK Tracing disabled\r\n", Tag );
@@ -560,7 +561,7 @@
     }
 
     snprintf( SendBuf, BufLen, "\n\n-----> C= %d %s PROXY: user tracing enabled.\n", (int)time(0), TraceUser );
-    write( Tracefd, SendBuf, strlen( SendBuf ) );
+    (void) !write( Tracefd, SendBuf, strlen( SendBuf ) );
     
     UnLockMutex( &trace );
     return( 0 );
@@ -1290,8 +1291,8 @@
 	    {
 		snprintf( TraceBuf, sizeof TraceBuf - 1, "\n\n-----> C= %d %s SERVER: sd [%d]\n",
 		    (int)time(0), ( (*TraceUser) ? TraceUser : "Null username" ), Server->conn->sd );
-		write( Tracefd, TraceBuf, strlen( TraceBuf ) );
-		write( Tracefd, Server->ReadBuf, status );
+		(void) !write( Tracefd, TraceBuf, strlen( TraceBuf ) );
+		(void) !write( Tracefd, Server->ReadBuf, status );
 	    }
 	    
 	    /* whatever we read from the server, ship off to the client */
@@ -1336,8 +1337,8 @@
 		if ( Client->TraceOn )
 		{
 		    snprintf( TraceBuf, sizeof TraceBuf - 1, "\n\n-----> C= %d %s CLIENT: sd [%d]\n", (int)time(0), ( (*TraceUser) ? TraceUser : "Null username" ), Client->conn->sd );
-		    write( Tracefd, TraceBuf, strlen( TraceBuf ) );
-		    write( Tracefd, Client->ReadBuf, status );
+		    (void) !write( Tracefd, TraceBuf, strlen( TraceBuf ) );
+		    (void) !write( Tracefd, Client->ReadBuf, status );
 		}
 		
 	    
@@ -1498,8 +1499,8 @@
 		if ( Server->TraceOn )
 		{
 		    snprintf( TraceBuf, sizeof TraceBuf - 1, "\n\n-----> C= %d %s SERVER: sd [%d]\n", (int)time(0), ( (*TraceUser) ? TraceUser : "Null username" ), Server->conn->sd );
-		    write( Tracefd, TraceBuf, strlen( TraceBuf ) );
-		    write( Tracefd, Server->ReadBuf, status );
+		    (void) !write( Tracefd, TraceBuf, strlen( TraceBuf ) );
+		    (void) !write( Tracefd, Server->ReadBuf, status );
 		}
 
 		if ( Server->ReadBuf[0] != '+' )
@@ -1533,8 +1534,8 @@
 		if ( Client->TraceOn )
 		{
 		    snprintf( TraceBuf, sizeof TraceBuf - 1, "\n\n-----> C= %d %s CLIENT: sd [%d]\n", (int)time(0), ( (*TraceUser) ? TraceUser : "Null username" ), Client->conn->sd );
-		    write( Tracefd, TraceBuf, strlen( TraceBuf ) );
-		    write( Tracefd, Client->ReadBuf, status );
+		    (void) !write( Tracefd, TraceBuf, strlen( TraceBuf ) );
+		    (void) !write( Tracefd, Client->ReadBuf, status );
 		}
 		
 		/* send any literal data back to the server */
