--- configure.ac.dist	2015-02-11 01:41:54.000000000 +0000
+++ configure.ac	2015-02-11 01:43:32.000000000 +0000
@@ -147,7 +147,7 @@
     CFLAGS="$CFLAGS -Wall -Wcast-align -Wshadow -Wstrict-prototypes"
     CPPFLAGS="$CPPFLAGS -Wall"
 
-    AC_MSG_CHECKING([if gcc supports -Wno-pointer-sign])
+    AC_MSG_CHECKING([if cc supports -Wno-pointer-sign])
     _gcc_cflags_save=$CFLAGS
     CFLAGS="-Wno-pointer-sign"
     AC_COMPILE_IFELSE([AC_LANG_PROGRAM([],[])],_gcc_psign=yes,_gcc_psign=no)
@@ -156,6 +156,16 @@
     if test x"$_gcc_psign" = xyes ; then
        CFLAGS="$CFLAGS -Wno-pointer-sign"
     fi
+
+    AC_MSG_CHECKING([if cc supports -Wno-implicit-function-declaration])
+    _gcc_cflags_save=$CFLAGS
+    CFLAGS="-Wno-implicit-function-declaration"
+    AC_COMPILE_IFELSE([AC_LANG_PROGRAM([],[])],_gcc_psign=yes,_gcc_psign=no)
+    AC_MSG_RESULT($_gcc_psign)
+    CFLAGS=$_gcc_cflags_save;
+    if test x"$_gcc_psign" = xyes ; then
+       CFLAGS="$CFLAGS -Wno-implicit-function-declaration"
+    fi
 fi
 
 
