--- a/configure.ac
+++ b/configure.ac
@@ -192,7 +192,7 @@ if test "$GCC" = yes; then
     if test x"$_gcc_silent_wno" = xyes ; then
       _gcc_warn=yes
     else
-      AC_MSG_CHECKING([if gcc supports -Wno-pointer-sign])
+      AC_MSG_CHECKING([if cc supports -Wno-pointer-sign])
       _gcc_cflags_save=$CFLAGS
       CFLAGS="-Wno-pointer-sign"
       AC_COMPILE_IFELSE([AC_LANG_PROGRAM([],[])],[_gcc_warn=yes],[_gcc_warn=no])
@@ -203,7 +203,7 @@ if test "$GCC" = yes; then
        CFLAGS="$CFLAGS -Wno-pointer-sign"
     fi
 
-    AC_MSG_CHECKING([if gcc supports -Wpointer-arith])
+    AC_MSG_CHECKING([if cc supports -Wpointer-arith])
     _gcc_cflags_save=$CFLAGS
     CFLAGS="-Wpointer-arith"
     AC_COMPILE_IFELSE([AC_LANG_PROGRAM([],[])],_gcc_warn=yes,_gcc_warn=no)
@@ -212,6 +212,16 @@ if test "$GCC" = yes; then
     if test x"$_gcc_warn" = xyes ; then
        CFLAGS="$CFLAGS -Wpointer-arith"
     fi
+
+    AC_MSG_CHECKING([if cc supports -Wno-implicit-function-declaration])
+    _gcc_cflags_save=$CFLAGS
+    CFLAGS="-Wno-implicit-function-declaration"
+    AC_COMPILE_IFELSE([AC_LANG_PROGRAM([],[])],_gcc_warn=yes,_gcc_warn=no)
+    AC_MSG_RESULT($_gcc_warn)
+    CFLAGS=$_gcc_cflags_save;
+    if test x"$_gcc_warn" = xyes ; then
+       CFLAGS="$CFLAGS -Wno-implicit-function-declaration"
+    fi
 fi
 
 
