--- motp-manager.dist	2011-08-15 15:37:00.768192290 +0100
+++ motp-manager	2011-08-15 15:38:21.207856229 +0100
@@ -144,7 +144,7 @@ user_add() {
 	if [[ ${#secret} -gt 32 || ${#secret} -lt 16 ]]
 	 then
 		echo "Please specify secret between 16 and 32 characters."
-	 elif [[ $secret != +([0-9abcdffABCDEF]) ]]
+	 elif [[ $secret != +([0-9abcdefABCDEF]) ]]
 	  then
 		echo "Specified secret is invalid HEX."
 	
