--- src/configure.in.dist       2014-07-26 10:58:37.059767039 +0100
+++ src/configure.in    2014-07-26 11:13:38.012836403 +0100
@@ -278,6 +278,11 @@ if test "$CRYPTO_IMPL" = builtin -a "x$e
        aesni_flags="-D__linux__ -f elf -m $aesni_machine"
        ;;
     esac
+    case "$host" in
+    *-gnux32)
+       aesni_flags="-D__linux__ -f elfx32 -m x32"
+       ;;
+    esac
     if test "x$aesni_obj" != x && test "x$aesni_flags" != x; then
        AC_CHECK_PROG(YASM,yasm,yasm)
        AC_CHECK_HEADERS(cpuid.h)
