diff --git a/drivers/regulator/fwnode_regulator.c b/drivers/regulator/fwnode_regulator.c
--- a/drivers/regulator/fwnode_regulator.c
+++ b/drivers/regulator/fwnode_regulator.c
@@ -711,27 +711,3 @@ struct regulator_dev *fwnode_parse_coupled_regulator(struct regulator_dev *rdev,
 
 	return c_rdev;
 }
-
-/*
- * Check if name is a supply name according to the '*-supply' pattern
- * return 0 if false
- * return length of supply name without the -supply
- */
-static int is_supply_name(const char *name)
-{
-	int strs, i;
-
-	strs = strlen(name);
-	/* string need to be at minimum len(x-supply) */
-	if (strs < 8)
-		return 0;
-	for (i = strs - 6; i > 0; i--) {
-		/* find first '-' and check if right part is supply */
-		if (name[i] != '-')
-			continue;
-		if (strcmp(name + i + 1, "supply") != 0)
-			return 0;
-		return i;
-	}
-	return 0;
-}
\ No newline at end of file
