diff --git a/drivers/clk/cix/clk-sky1-audss.c b/drivers/clk/cix/clk-sky1-audss.c
--- a/drivers/clk/cix/clk-sky1-audss.c
+++ b/drivers/clk/cix/clk-sky1-audss.c
@@ -1052,9 +1052,15 @@ static int sky1_audss_clks_get(struct sky1_audss_clks_priv *priv)
 
 	for (i = 0; i < devtype_data->clk_num; i++) {
 		priv->clks[i] = devm_clk_get(priv->dev, devtype_data->clk_names[i]);
-		if (IS_ERR(priv->clks[i]))
-			return dev_err_probe(priv->dev, PTR_ERR(priv->clks[i]),
+		if (IS_ERR(priv->clks[i])) {
+			int ret = PTR_ERR(priv->clks[i]);
+
+			if (ret == -ENOENT && has_acpi_companion(priv->dev))
+				ret = -EPROBE_DEFER;
+
+			return dev_err_probe(priv->dev, ret,
 					     "failed to get clock %s", devtype_data->clk_names[i]);
+		}
 	}
 
 	return 0;
@@ -1288,9 +1294,10 @@ static int sky1_audss_clk_probe(struct platform_device *pdev)
 		clk_hw_unregister_composite(clk_table[devtype_data->clk_cfg[i].id]);
 fail_fixed_clk:
 fail_clks_set:
-	pm_runtime_put_sync(dev);
 err_pm:
 	pm_runtime_disable(dev);
+	pm_runtime_set_suspended(dev);
+	pm_runtime_put_noidle(dev);
 	return ret;
 }
 
