
diff --git a/drivers/reset/core.c b/drivers/reset/core.c
--- a/drivers/reset/core.c
+++ b/drivers/reset/core.c
@@ -1238,12 +1238,18 @@ struct reset_control *__reset_control_get(struct device *dev, const char *id,
 	bool acquired = flags & RESET_CONTROL_FLAGS_BIT_ACQUIRED;
 	bool optional = flags & RESET_CONTROL_FLAGS_BIT_OPTIONAL;
 	struct fwnode_handle *fwnode = dev_fwnode(dev);
+	struct reset_control *rstc;
 
 	if (WARN_ON(shared && acquired))
 		return ERR_PTR(-EINVAL);
 
-	if (fwnode)
-		return __fwnode_reset_control_get(fwnode, id, index, flags);
+	if (fwnode) {
+		rstc = __fwnode_reset_control_get(fwnode, id, index, flags);
+		if (!IS_ERR_OR_NULL(rstc) || !has_acpi_companion(dev))
+			return rstc;
+		if (IS_ERR(rstc) && PTR_ERR(rstc) != -ENOENT)
+			return rstc;
+	}
 
 	return optional ? NULL : ERR_PTR(-ENOENT);
 }
