diff --git a/drivers/media/platform/cix/armcb-isp/sensor/actuator/armcb_actuator.h b/drivers/media/platform/cix/armcb-isp/sensor/actuator/armcb_actuator.h
--- a/drivers/media/platform/cix/armcb-isp/sensor/actuator/armcb_actuator.h
+++ b/drivers/media/platform/cix/armcb-isp/sensor/actuator/armcb_actuator.h
@@ -22,7 +22,6 @@
 #include <linux/gpio.h>
 #include <linux/media.h>
 #include <linux/module.h>
-#include <linux/of_gpio.h>
 #include <linux/of_graph.h>
 #include <linux/regulator/consumer.h>
 #include <linux/slab.h>
