diff --git a/drivers/media/platform/cix/armcb-isp/sensor/armcb_sensor.c b/drivers/media/platform/cix/armcb-isp/sensor/armcb_sensor.c
--- a/drivers/media/platform/cix/armcb-isp/sensor/armcb_sensor.c
+++ b/drivers/media/platform/cix/armcb-isp/sensor/armcb_sensor.c
@@ -579,8 +579,13 @@ static int imgsens_probe(struct i2c_client *client)
 	else
 		pimgsens_sd->imgs_inst.mclk = devm_clk_get_optional(&client->dev, "mclk");
 
-	if (IS_ERR_OR_NULL(pimgsens_sd->imgs_inst.mclk))
-		LOG(LOG_ERR, "failed to get cam mclk\n");
+	if (IS_ERR(pimgsens_sd->imgs_inst.mclk)) {
+		res = dev_err_probe(dev, PTR_ERR(pimgsens_sd->imgs_inst.mclk),
+				    "failed to get camera mclk\n");
+		goto EXIT_RET;
+	}
+	if (!pimgsens_sd->imgs_inst.mclk)
+		dev_dbg(dev, "optional camera mclk not described\n");
 
 	LOG(LOG_INFO, "scuess - ");
 	return res;
