--- init.d/devfs.in.dist
+++ init.d/devfs.in
@@ -79,11 +80,11 @@
 	# Mount required directories as user may not have them in /etc/fstab
 	for x in \
 		"mqueue /dev/mqueue 1777 ,nodev mqueue" \
-		"devpts /dev/pts 0755 ,gid=5,mode=0620 devpts" \
+		"devpts /dev/pts 0755 ,gid=5,mode=0620,ptmxmode=0666 devpts" \
 		"tmpfs /dev/shm 1777 ,nodev,mode=1777 shm" \
 	; do
 		set -- $x
-		grep -Eq "[[:space:]]+$1$" /proc/filesystems || continue
+		grep -q "\s\+${1}$" /proc/filesystems || continue
 		mountinfo -q $2 && continue
 
 		if [ ! -d $2 ]; then
@@ -99,6 +100,12 @@
 			eend $?
 		fi
 	done
+	[ -e /dev/pts/ptmx ] && \
+		[ ! -L /dev/ptmx ] && \
+		ln -s pts/ptmx /dev/ptmx.new && \
+		{
+			mv /dev/ptmx.new /dev/ptmx || rm /dev/ptmx.new
+		}
 }
 
 restorecon_dev()
