--- init.d/swap.in.dist	2013-06-29 04:25:00.867872169 +0100
+++ init.d/swap.in	2013-06-30 23:28:40.996168111 +0100
@@ -26,6 +26,21 @@
 	# Try to unmount all tmpfs filesystems not in use, else a deadlock may
 	# occur. As $RC_SVCDIR may also be tmpfs we cd to it to lock it
 	cd "$RC_SVCDIR"
+	# ... and also keep any tmpfs filesystems used by /etc/init.d/tmpfs...
+	if [[ -r "/etc/conf.d/tmpfs" && -s "/etc/conf.d/tmpfs" ]]; then
+		local FS
+		for FS in $(
+			  grep -E "_FSTAB" "/etc/conf.d/tmpfs" \
+			| sed 's/#.*$//' \
+			| grep -v '^\s*$' \
+			| sed -r 's|^\s*[a-zA-Z_][a-zA-Z0-9_]*_FSTAB=["'\''][a-zA-Z_][a-zA-Z0-9_]*\s(/[^\s]*)\stmpfs.*["'\'']\s*$|\1|' \
+			| grep -v '='
+		); do
+			[[ -d "$FS" ]] && \
+				flock -xn "$FS" -c 'sleep 5' >/dev/null 2>&1
+		done
+		unset FS
+	fi
 	umount -a -t tmpfs 2>/dev/null
 
 	case "$RC_UNAME" in
--- init.d/swapfiles.in.dist	2013-06-29 04:24:47.194503514 +0100
+++ init.d/swapfiles.in	2013-06-30 23:29:07.446256824 +0100
@@ -27,6 +27,21 @@
 	# fixme: Do we need this here since we are only unmounting swap files
 	# and loopback swap?
 	cd "$RC_SVCDIR"
+	# ... and also keep any tmpfs filesystems used by /etc/init.d/tmpfs...
+	if [[ -r "/etc/conf.d/tmpfs" && -s "/etc/conf.d/tmpfs" ]]; then
+		local FS
+		for FS in $(
+			  grep -E "_FSTAB" "/etc/conf.d/tmpfs" \
+			| sed 's/#.*$//' \
+			| grep -v '^\s*$' \
+			| sed -r 's|^\s*[a-zA-Z_][a-zA-Z0-9_]*_FSTAB=["'\''][a-zA-Z_][a-zA-Z0-9_]*\s(/[^\s]*)\stmpfs.*["'\'']\s*$|\1|' \
+			| grep -v '='
+		); do
+			[[ -d "$FS" ]] && \
+				flock -xn "$FS" -c 'sleep 5' >/dev/null 2>&1
+		done
+		unset FS
+	fi
 	umount -a -t tmpfs 2>/dev/null
 
 	case "$RC_UNAME" in
