--- etc/rc.conf.dist
+++ etc/rc.conf
@@ -17,7 +17,7 @@
 
 # If we need to drop to a shell, you can specify it here.
 # If not specified we use $SHELL, otherwise the one specified in /etc/passwd,
-# otherwise /bin/sh
+# otherwise /bin/sh.
 # Linux users could specify /sbin/sulogin
 #rc_shell=/bin/sh
 
@@ -48,7 +48,7 @@
 # /var/log/rc.log
 # NOTE: Linux systems require the devfs service to be started before
 # logging can take place and as such cannot log the sysinit runlevel.
-#rc_logger="YES"
+#rc_logger="NO"
 
 # Through rc_log_path you can specify a custom log file.
 # The default value is: /var/log/rc.log
@@ -92,7 +92,7 @@
 #unicode="NO"
 
 # This is how long fuser should wait for a remote server to respond. The
-# default is 60 seconds, but  it can be adjusted here.
+# default is 60 seconds, but it can be adjusted here.
 #rc_fuser_timeout=60
 
 # Below is the default list of network fstypes.
