--- sh/rc-cgroup.sh.in.dist	2014-03-08 17:07:53.519310178 +0000
+++ sh/rc-cgroup.sh.in	2014-03-08 17:09:11.739520648 +0000
@@ -84,7 +84,11 @@ cgroup_add_service()
 	if [ -d "$openrc_cgroup" ]; then
 		cgroup="$openrc_cgroup/$RC_SVCNAME"
 		mkdir -p "$cgroup"
-		[ -f "$cgroup/tasks" ] && echo 0 > "$cgroup/tasks"
+                # There is a race-condition here for fast-to-exit services,
+                # such as 'local'.  Rather than take any specific action,
+                # we'll settle for throwing an error...
+                [ -f "$cgroup/tasks" ] && \
+                        echo 0 > "$cgroup/tasks" 2>/dev/null
 	fi
 }
 
