--- sh/tmpfiles.sh.in.dist	2013-06-13 00:22:23.848763196 +0100
+++ sh/tmpfiles.sh.in	2013-06-13 00:28:33.117474259 +0100
@@ -191,7 +191,8 @@
 FILE=
 fragments=
 # XXX: The harcoding of /usr/lib/ is an explicit choice by upstream
-tmpfiles_dirs='/usr/lib/tmpfiles.d/ /etc/tmpfiles.d/ /run/tmpfiles.d/'
+#tmpfiles_dirs='/usr/lib/tmpfiles.d/ /etc/tmpfiles.d/ /run/tmpfiles.d/'
+tmpfiles_dirs='/usr/lib/tmpfiles.d/ /etc/tmpfiles.d/ /var/run/tmpfiles.d/'
 tmpfiles_basenames=''
 tmpfiles_d=''
 # Build a list of sorted unique basenames
--- sh/init.sh.Linux.in.dist	2013-06-13 00:21:56.743291837 +0100
+++ sh/init.sh.Linux.in	2013-06-13 00:25:07.246523005 +0100
@@ -3,6 +3,62 @@
 # Copyright (c) 2007-2009 Roy Marples <roy@marples.name>
 # Released under the 2-clause BSD license.
 
+# This basically mounts $RC_SVCDIR as a ramdisk.
+# The tricky part is finding something our kernel supports
+# tmpfs and ramfs are easy, so force one or the other.
+svcdir_restorecon()
+{
+	local rc=0
+	if [ -x /usr/sbin/selinuxenabled -a -c /selinux/null ] &&
+	  selinuxenabled; then
+		restorecon $RC_SVCDIR
+		rc=$?
+	fi
+	return $rc
+}
+
+mount_svcdir()
+{
+	# mount from fstab if we can
+	fstabinfo --mount "$RC_SVCDIR" && return 0
+
+	local fs= fsopts="-o rw,noexec,nodev,nosuid"
+	local svcsize=${rc_svcsize:-1024}
+
+	# Some buggy kernels report tmpfs even when not present :(
+	if grep -Eq "[[:space:]]+tmpfs$" /proc/filesystems; then
+		local tmpfsopts="${fsopts},mode=755,size=${svcsize}k"
+		mount -n -t tmpfs $tmpfsopts rc-svcdir "$RC_SVCDIR"
+		if [ $? -eq 0 ]; then
+			svcdir_restorecon
+			[ $? -eq 0 ] && return 0
+		fi
+	fi
+
+	if grep -Eq "[[:space:]]+ramfs$" /proc/filesystems; then
+		fs="ramfs"
+		# ramfs has no special options
+	elif [ -e /dev/ram0 ] \
+		&& grep -Eq "[[:space:]]+ext2$" /proc/filesystems; then
+		devdir="/dev/ram0"
+		fs="ext2"
+		dd if=/dev/zero of="$devdir" bs=1k count="$svcsize"
+		mkfs -t "$fs" -i 1024 -vm0 "$devdir" "$svcsize"
+	else
+		echo
+		eerror "OpenRC requires tmpfs, ramfs or a ramdisk + ext2"
+		eerror "compiled into the kernel"
+		echo
+		return 1
+	fi
+
+	mount -n -t "$fs" $fsopts rc-svcdir "$RC_SVCDIR"
+	if [ $? -eq 0 ]; then
+		svcdir_restorecon
+		[ $? -eq 0 ] && return 0
+	fi
+}
+
 . "$RC_LIBEXECDIR"/sh/functions.sh
 [ -r /etc/rc.conf ] && . /etc/rc.conf
 
@@ -46,8 +102,41 @@
 		fi
 		mkdir /run
 	else
-		eerror "The /run directory does not exist. Unable to continue."
-		return 1
+		#eerror "The /run directory does not exist. Unable to continue."
+		#return 1
+		:
+		# awk lives in /usr...
+		if type -pf awk >/dev/null 2>&1; then
+			boot_services="$( rc-status -Ca 2>/dev/null | awk 'BEGIN { doprint = 0 } ; /^Runlevel: / { doprint = 0 ; ( "boot" == $2 || "sysinit" == $2 ) && doprint = 2 } ; ( 1 == doprint ) { print $1 } ; ( 2 == doprint ) { doprint = 1 }' )"
+		else
+			boot_services="$( rc-status -Ca 2>/dev/null | sed -r '/^Runlevel: / s/^(.*)$/Runlevel: \n\1/' | sed -nr '/^Runlevel: (sysinit|boot)/,/^(Dynamic )?Runlevel: /p' | grep -v '^Runlevel:' | sed 's/^ \+// ; s/ .*$//' )"
+		fi
+
+		if echo ${boot_services:-lvm} | grep -Eq "dmeventd|lvm|device-mapper" >/dev/null 2>&1 && [ -d /var -a ! -w /var ]; then
+			if ! mountinfo -q /var/run; then
+				ebegin "Mounting /var/run"
+				rc=0
+				if ! fstabinfo --mount /var/run; then
+					mount -t tmpfs -o mode=0755,nosuid,nodev tmpfs /var/run
+					rc=$?
+				fi
+				if [ $rc != 0 ]; then
+					eerror "Unable to mount tmpfs on /var/run."
+				fi
+			fi
+			if ! mountinfo -q /var/lock; then
+				ebegin "Mounting /var/lock"
+				rc=0
+				if ! fstabinfo --mount /var/lock; then
+					mount -t tmpfs -o mode=0755,nosuid,nodev tmpfs /var/lock
+					rc=$?
+					checkpath -d -m 0775 -o root:uucp /var/lock
+				fi
+				if [ $rc != 0 ]; then
+					eerror "Unable to mount tmpfs on /var/lock"
+				fi
+			fi
+		fi
 	fi
 fi
 
@@ -55,24 +144,28 @@
 	rm /run/openrc
 fi
 
-if [ "$sys" = VSERVER ]; then
-	rm -rf /run/*
-elif ! mountinfo -q /run; then
-	ebegin "Mounting /run"
-	rc=0
-	if ! fstabinfo --mount /run; then
-		mount -t tmpfs -o mode=0755,nosuid,nodev tmpfs /run
-		rc=$?
-	fi
-	if [ $rc != 0 ]; then
-		eerror "Unable to mount tmpfs on /run."
-		eerror "Can't continue."
-		exit 1
+if [ -d /run ]; then
+	if [ "$sys" = VSERVER ]; then
+		rm -rf /run/*
+	elif ! mountinfo -q /run; then
+		ebegin "Mounting /run"
+		rc=0
+		if ! fstabinfo --mount /run; then
+			mount -t tmpfs -o mode=0755,nosuid,nodev tmpfs /run
+			rc=$?
+		fi
+		if [ $rc != 0 ]; then
+			eerror "Unable to mount tmpfs on /run."
+			eerror "Can't continue."
+			exit 1
+		fi
 	fi
+	checkpath -d -m 0775 -o root:uucp /run/lock
+else
+	[ -w /var ] && checkpath -d -m 0775 -o root:uucp /var/lock
 fi
 
 checkpath -d $RC_SVCDIR
-checkpath -d -m 0775 -o root:uucp /run/lock
 
 # Try to mount xenfs as early as possible, otherwise rc_sys() will always
 # return RC_SYS_XENU and will think that we are in a domU while it's not.
@@ -84,9 +177,4 @@
 	eend $?
 fi
 
-if [ -e "$RC_LIBEXECDIR"/cache/deptree ]; then
-	cp -p "$RC_LIBEXECDIR"/cache/* "$RC_SVCDIR" 2>/dev/null
-fi
-
-echo sysinit >"$RC_SVCDIR"/softlevel
-exit 0
+. "$RC_LIBEXECDIR"/sh/init-common-post.sh
--- init.d/bootmisc.in.dist	2013-06-13 00:22:57.503945672 +0100
+++ init.d/bootmisc.in	2013-06-13 00:31:10.160571880 +0100
@@ -112,10 +112,10 @@
 		fi
 	done
 
-	if [ "$RC_UNAME" = Linux -a -d /run ]; then
-		migrate_to_run	/var/lock /run/lock
-		migrate_to_run	/var/run /run
-	fi
+	#if [ "$RC_UNAME" = Linux -a -d /run ]; then
+	#	migrate_to_run	/var/lock /run/lock
+	#	migrate_to_run	/var/run /run
+	#fi
 
 	if checkpath -W /var/run; then
 		ebegin "Creating user login records"
@@ -145,7 +145,7 @@
 			case "$x" in
 				*.pid)
 					start-stop-daemon --test --quiet \
-					--stop --pidfile "$x" && continue
+					--stop --pidfile "$x" >/dev/null 2>&1 && continue
 				;;
 			esac
 			rm -f -- "$x"
--- init.d/procfs.in.dist	2013-06-13 00:23:09.275041544 +0100
+++ init.d/procfs.in	2013-06-13 00:31:40.345911835 +0100
@@ -30,6 +30,7 @@
 				local fmts
 				ebegin "Loading custom binary format handlers"
 				fmts=$(grep -hsv -e '^[#;]' -e '^[[:space:]]*$' \
+					/var/run/binfmt.d/*.conf \
 					/run/binfmt.d/*.conf \
 					@SYSCONFDIR@/binfmt.d/*.conf \
 					""/usr/lib/binfmt.d/*.conf)
--- src/librc/rc.h.in.dist	2013-06-13 00:23:47.400032495 +0100
+++ src/librc/rc.h.in	2013-06-13 00:25:07.199858389 +0100
@@ -39,8 +39,10 @@
 #define RC_LIBEXECDIR           "@LIBEXECDIR@"
 #if defined(PREFIX)
 #define RC_SVCDIR               RC_LIBEXECDIR "/init.d"
+/*
 #elif defined(__linux__)
 #define RC_SVCDIR               "/run/openrc"
+ */
 #else
 #define RC_SVCDIR               RC_LIBEXECDIR "/init.d"
 #endif
