diff --git a/drivers/usb/cdns3/cdns3-plat.c b/drivers/usb/cdns3/cdns3-plat.c
--- a/drivers/usb/cdns3/cdns3-plat.c
+++ b/drivers/usb/cdns3/cdns3-plat.c
@@ -216,6 +216,7 @@ static int cdns3_controller_suspend(struct device *dev, pm_message_t msg)
 	struct cdns *cdns = dev_get_drvdata(dev);
 	bool wakeup;
 	unsigned long flags;
+	int ret;
 
 	if (cdns->in_lpm)
 		return 0;
@@ -225,7 +226,10 @@ static int cdns3_controller_suspend(struct device *dev, pm_message_t msg)
 	else
 		wakeup = device_may_wakeup(dev);
 
-	cdns3_set_platform_suspend(cdns->dev, true, wakeup);
+	ret = cdns3_set_platform_suspend(cdns->dev, true, wakeup);
+	if (ret)
+		return ret;
+
 	set_phy_power_off(cdns);
 	spin_lock_irqsave(&cdns->lock, flags);
 	cdns->in_lpm = true;
@@ -260,10 +264,15 @@ static int cdns3_controller_resume(struct device *dev, pm_message_t msg)
 	if (ret)
 		return ret;
 
-	cdns3_set_platform_suspend(cdns->dev, false, false);
+	ret = cdns3_set_platform_suspend(cdns->dev, false, false);
+	if (ret)
+		goto err_power_off_phys;
+
+	ret = cdns_resume(cdns);
+	if (ret)
+		goto err_enter_low_power;
 
 	spin_lock_irqsave(&cdns->lock, flags);
-	cdns_resume(cdns);
 	cdns->in_lpm = false;
 	spin_unlock_irqrestore(&cdns->lock, flags);
 	cdns_set_active(cdns, !PMSG_IS_AUTO(msg));
@@ -273,6 +282,12 @@ static int cdns3_controller_resume(struct device *dev, pm_message_t msg)
 	}
 	dev_dbg(cdns->dev, "%s ends\n", __func__);
 
+	return 0;
+
+err_enter_low_power:
+	cdns3_set_platform_suspend(cdns->dev, true, false);
+err_power_off_phys:
+	set_phy_power_off(cdns);
 	return ret;
 }
 
@@ -293,11 +308,15 @@ static int cdns3_plat_suspend(struct device *dev)
 	struct cdns *cdns = dev_get_drvdata(dev);
 	int ret;
 
-	cdns_suspend(cdns);
+	ret = cdns_suspend(cdns);
+	if (ret)
+		return ret;
 
 	ret = cdns3_controller_suspend(dev, PMSG_SUSPEND);
-	if (ret)
+	if (ret) {
+		cdns_resume(cdns);
 		return ret;
+	}
 
 	if (device_may_wakeup(dev) && cdns->wakeup_irq)
 		enable_irq_wake(cdns->wakeup_irq);
diff --git a/drivers/usb/cdns3/cdnsp-pci.c b/drivers/usb/cdns3/cdnsp-pci.c
--- a/drivers/usb/cdns3/cdnsp-pci.c
+++ b/drivers/usb/cdns3/cdnsp-pci.c
@@ -204,15 +204,15 @@ static int __maybe_unused cdnsp_pci_suspend(struct device *dev)
 static int __maybe_unused cdnsp_pci_resume(struct device *dev)
 {
 	struct cdns *cdns = dev_get_drvdata(dev);
-	unsigned long flags;
 	int ret;
 
-	spin_lock_irqsave(&cdns->lock, flags);
 	ret = cdns_resume(cdns);
-	spin_unlock_irqrestore(&cdns->lock, flags);
+	if (ret)
+		return ret;
+
 	cdns_set_active(cdns, 1);
 
-	return ret;
+	return 0;
 }
 
 static const struct dev_pm_ops cdnsp_pci_pm_ops = {
diff --git a/drivers/usb/cdns3/core.c b/drivers/usb/cdns3/core.c
--- a/drivers/usb/cdns3/core.c
+++ b/drivers/usb/cdns3/core.c
@@ -508,17 +508,18 @@ int cdns_suspend(struct cdns *cdns)
 {
 	struct device *dev = cdns->dev;
 	unsigned long flags;
+	int ret = 0;
 
 	if (pm_runtime_status_suspended(dev))
 		pm_runtime_resume(dev);
 
 	if (cdns->roles[cdns->role]->suspend) {
 		spin_lock_irqsave(&cdns->lock, flags);
-		cdns->roles[cdns->role]->suspend(cdns, false);
+		ret = cdns->roles[cdns->role]->suspend(cdns, false);
 		spin_unlock_irqrestore(&cdns->lock, flags);
 	}
 
-	return 0;
+	return ret;
 }
 EXPORT_SYMBOL_GPL(cdns_suspend);
 
@@ -552,7 +553,7 @@ int cdns_resume(struct cdns *cdns)
 	}
 
 	if (!role_changed && cdns->roles[cdns->role]->resume)
-		cdns->roles[cdns->role]->resume(cdns, power_lost);
+		return cdns->roles[cdns->role]->resume(cdns, power_lost);
 
 	return 0;
 }
