diff --git a/drivers/firmware/arm_scmi/driver.c b/drivers/firmware/arm_scmi/driver.c
--- a/drivers/firmware/arm_scmi/driver.c
+++ b/drivers/firmware/arm_scmi/driver.c
@@ -32,6 +32,7 @@
 #include <linux/of.h>
 #include <linux/platform_device.h>
 #include <linux/processor.h>
+#include <linux/property.h>
 #include <linux/refcount.h>
 #include <linux/slab.h>
 #include <linux/xarray.h>
@@ -2757,7 +2758,7 @@ scmi_txrx_setup(struct scmi_info *info, struct device_node *of_node,
 			ret = 0;
 	}
 
-	if (ret)
+	if (ret && ret != -EPROBE_DEFER)
 		dev_err(info->dev,
 			"failed to setup channel for protocol:0x%X\n", prot_id);
 
@@ -2976,19 +2977,33 @@ static struct scmi_debug_info *scmi_debugfs_common_setup(struct scmi_info *info)
 	char top_dir[16];
 	struct dentry *trans, *top_dentry;
 	struct scmi_debug_info *dbg;
-	const char *c_ptr = NULL;
+	const char *c_ptr = NULL, *name;
 
 	dbg = devm_kzalloc(info->dev, sizeof(*dbg), GFP_KERNEL);
 	if (!dbg)
 		return NULL;
 
-	dbg->name = kstrdup(of_node_full_name(info->dev->of_node), GFP_KERNEL);
+	if (info->dev->of_node) {
+		name = of_node_full_name(info->dev->of_node);
+		of_property_read_string(info->dev->of_node,
+					"compatible", &c_ptr);
+	} else {
+		struct fwnode_handle *fwnode = dev_fwnode(info->dev);
+
+		name = dev_name(info->dev);
+		if (fwnode)
+			c_ptr = fwnode_get_name(fwnode);
+	}
+
+	if (!c_ptr)
+		c_ptr = dev_driver_string(info->dev);
+
+	dbg->name = kstrdup(name, GFP_KERNEL);
 	if (!dbg->name) {
 		devm_kfree(info->dev, dbg);
 		return NULL;
 	}
 
-	of_property_read_string(info->dev->of_node, "compatible", &c_ptr);
 	dbg->type = kstrdup(c_ptr, GFP_KERNEL);
 	if (!dbg->type) {
 		kfree(dbg->name);
