diff --git a/drivers/mfd/syscon.c b/drivers/mfd/syscon.c
--- a/drivers/mfd/syscon.c
+++ b/drivers/mfd/syscon.c
@@ -12,6 +12,7 @@
 #include <linux/cleanup.h>
 #include <linux/clk.h>
 #include <linux/err.h>
+#include <linux/device/bus.h>
 #include <linux/hwspinlock.h>
 #include <linux/list.h>
 #include <linux/mutex.h>
@@ -369,18 +370,25 @@ struct regmap *device_syscon_regmap_lookup_by_property(struct device *dev,
 						       const char *property)
 {
 	struct fwnode_handle *fwnode;
-	struct platform_device *pdev;
+	struct device *sdev;
 	struct syscon *syscon;
 
 	fwnode = fwnode_find_reference(dev_fwnode(dev), property, 0);
-	if (IS_ERR_OR_NULL(fwnode))
+	if (IS_ERR(fwnode))
+		return ERR_CAST(fwnode);
+	if (!fwnode)
 		return ERR_PTR(-ENODEV);
 
-	pdev = to_platform_device(fwnode->dev);
-	if (!pdev->dev.driver_data)
-		return ERR_PTR(-ENOENT);
-	syscon = platform_get_drvdata(pdev);
+	sdev = bus_find_device_by_fwnode(&platform_bus_type, fwnode);
+	fwnode_handle_put(fwnode);
+	if (!sdev)
+		return ERR_PTR(-EPROBE_DEFER);
+
+	syscon = dev_get_drvdata(sdev);
+	put_device(sdev);
 	if (!syscon)
+		return ERR_PTR(-EPROBE_DEFER);
+	if (!syscon->regmap)
 		return ERR_PTR(-ENODEV);
 
 	return syscon->regmap;
