--- ./upnp/Makefile.am.dist	2016-07-27 21:59:56.865497461 +0100
+++ ./upnp/Makefile.am	2016-07-27 22:01:46.667202854 +0100
@@ -7,6 +7,7 @@
 SUBDIRS = doc . sample unittest
 
 AM_CPPFLAGS = \
+	-include $(top_builddir)/autoconfig.h \
 	-I$(srcdir)/inc \
 	-I$(top_srcdir)/threadutil/inc \
 	-I$(top_srcdir)/ixml/inc
--- ./upnp/src/api/UpnpString.c.dist	2016-07-27 21:59:47.625634453 +0100
+++ ./upnp/src/api/UpnpString.c	2016-07-27 22:01:22.550893740 +0100
@@ -28,34 +28,6 @@
 	/* Other systems have strncasecmp */
 #endif
 
-#ifndef UPNP_USE_MSVCPP
-	/* VC has strnlen which is already included but with (potentially) different linkage */
-	/* strnlen() is a GNU extension. */
-	#if !HAVE_STRNLEN
-		static size_t strnlen(const char *s, size_t n)
-		{
-			const char *p = (const char *)memchr(s, 0, n);
-			return p ? p - s : n;
-		}
-	#endif /* !HAVE_STRNLEN */
-#endif /* WIN32 */
-
-/* strndup() is a GNU extension. */
-#if !HAVE_STRNDUP || defined(WIN32)
-	static char *strndup(const char *__string, size_t __n)
-	{
-		size_t strsize = strnlen(__string, __n);
-		char *newstr = (char *)malloc(strsize + 1);
-		if (newstr == NULL)
-			return NULL;
-
-		strncpy(newstr, __string, strsize);
-		newstr[strsize] = 0;
-
-		return newstr;
-	}
-#endif /* HAVE_STRNDUP && !defined(WIN32) */
-
 /*!
  * \brief Internal implementation of the class UpnpString.
  *
--- ./configure.ac.dist	2016-07-27 21:53:28.914582627 +0100
+++ ./configure.ac	2016-07-27 21:56:01.302323296 +0100
@@ -7,9 +7,9 @@
 # (C) Copyright 2005-2007 Rémi Turboult <r3mi@users.sourceforge.net>
 #
 
-AC_PREREQ(2.60)
+AC_PREREQ([2.60])
 
-AC_INIT([libupnp], [1.8.0], [mroberto@users.sourceforge.net])
+AC_INIT([libupnp],[1.8.0],[mroberto@users.sourceforge.net])
 dnl ############################################################################
 dnl # *Independently* of the above libupnp package version, the libtool version
 dnl # of the 3 libraries need to be updated whenever there is a change released:
@@ -371,7 +371,6 @@ AC_CANONICAL_HOST
 #    installed libraries.
 #
 AC_CONFIG_HEADERS([autoconfig.h upnp/inc/upnpconfig.h])
-#AC_SYS_LARGEFILE_SENSITIVE
 
 AC_REVISION([$Revision: 1.11 $])
 
@@ -525,10 +524,12 @@ AC_MSG_RESULT($docdir)
 #
 # Checks for programs
 #
-AC_PROG_CC
+AC_PROG_CC_STDC
+AC_USE_SYSTEM_EXTENSIONS
+AC_SYS_LARGEFILE
 AM_PROG_CC_C_O
 m4_ifdef([AM_PROG_AR], [AM_PROG_AR])
-AC_PROG_LIBTOOL
+LT_INIT
 AC_PROG_INSTALL
 AC_PROG_MAKE_SET
 AC_PROG_EGREP
@@ -552,9 +553,6 @@ freebsd*)
 		#:
 		# Use -O0 in debug so that variables do not get optimized out
 		AX_CFLAGS_GCC_OPTION([-O0, -g])
-	else
-		# add optimise for size
-		AX_CFLAGS_GCC_OPTION([-Os])
 	fi
 	;;
 esac
@@ -568,8 +566,6 @@ echo "----------------------------------
 #
 AC_TYPE_SIZE_T
 AC_TYPE_OFF_T
-AC_DEFINE([_LARGE_FILE_SOURCE], [], [Large files support])
-AC_DEFINE([_FILE_OFFSET_BITS], [64], [File Offset size])
 
 
 #
--- ./ixml/Makefile.am.dist	2016-07-27 21:56:12.028830930 +0100
+++ ./ixml/Makefile.am	2016-07-27 21:58:34.903379314 +0100
@@ -7,8 +7,9 @@
 
 SUBDIRS 		= doc
 
-AM_CPPFLAGS		= -I$(srcdir)/inc -I$(srcdir)/src/inc \
-                          -I$(top_srcdir)/upnp/inc
+AM_CPPFLAGS		= -include $(top_builddir)/autoconfig.h \
+			  -I$(srcdir)/inc -I$(srcdir)/src/inc \
+			  -I$(top_srcdir)/upnp/inc
 AM_CFLAGS		= 
 
 LDADD			= libixml.la
--- ./threadutil/Makefile.am.dist	2016-07-27 21:58:50.983140914 +0100
+++ ./threadutil/Makefile.am	2016-07-27 21:59:19.306054324 +0100
@@ -4,8 +4,9 @@
 # (C) Copyright 2005 Remi Turboult <r3mi@users.sourceforge.net>
 #
 
-AM_CPPFLAGS		= -I$(srcdir)/inc -I$(srcdir)/src/inc \
-                          -I$(top_srcdir)/upnp/inc
+AM_CPPFLAGS		= -include $(top_builddir)/autoconfig.h \
+			  -I$(srcdir)/inc -I$(srcdir)/src/inc \
+			  -I$(top_srcdir)/upnp/inc
 
 if ENABLE_DEBUG
     AM_CPPFLAGS		+= -DDEBUG -DSTATS
