diff --git a/drivers/media/platform/cix/if/mvx_session.c b/drivers/media/platform/cix/if/mvx_session.c
--- a/drivers/media/platform/cix/if/mvx_session.c
+++ b/drivers/media/platform/cix/if/mvx_session.c
@@ -86,9 +86,6 @@ module_param(fw_watchdog_timeout, uint, 0444);
 static uint wait_pending_timeout = 3000; /* 3s should be enough in worst case - 32Kx32K decode */
 module_param(wait_pending_timeout, uint, 0444);
 
-static bool enable_buffer_dump = 0;
-module_param(enable_buffer_dump, bool, 0660);
-
 /****************************************************************************
  * Private functions and and variables
  ****************************************************************************/
@@ -557,73 +554,6 @@ static void print_debug(struct mvx_session *session)
         session->fw.ops.print_debug(&session->fw);
 }
 
-static void dump_ivf_header(struct mvx_session *session)
-{
-    struct mvx_session_port *input = &session->port[MVX_DIR_INPUT];
-    bool is_ivf = (input->format == MVX_FORMAT_AV1 ||
-                input->format == MVX_FORMAT_VP8 ||
-                input->format == MVX_FORMAT_VP9);
-
-    if (!current->fs) {
-        MVX_SESSION_ERR(session, "process exit abnormally,fs has been released!!!");
-        return;
-    }
-
-    if (is_ivf) {
-        uint32_t alloc_bytes = MVE_PAGE_SIZE;
-        struct mvx_session_port *output = &session->port[MVX_DIR_OUTPUT];
-        struct mvx_ivf_header ivf_header = {
-            .signature = v4l2_fourcc('D', 'K', 'I', 'F'),
-            .version = 0,
-            .length = sizeof(struct mvx_ivf_header),
-            .fourcc = input->pixelformat,
-            .width = output->width,
-            .height = output->height,
-            .frameRate = (30 << 16),
-            .timeScale = (1 << 16),
-            .frameCount = input->dump_count,
-            .padding = 0
-        };
-        uint8_t *ivf_header_buf = (uint8_t*)(&ivf_header);
-        struct file *ivf_fp = NULL;
-        char ivf_file[64];
-        char data_file[64];
-        scnprintf(ivf_file, sizeof(ivf_file) - 1,
-                        "/data/input_session_%p.ivf", session);
-        scnprintf(data_file, sizeof(data_file) - 1,
-                        "/data/input_session_%p.bin", session);
-
-        ivf_fp = filp_open(ivf_file, O_RDWR | O_CREAT, 0644);
-        if (IS_ERR(ivf_fp)) {
-            MVX_SESSION_WARN(session, "warning: open file(%s) fail", ivf_file);
-            return;
-        }
-        // write ivf header into new file at the beginning
-        kernel_write(ivf_fp, ivf_header_buf, sizeof(struct mvx_ivf_header), &(ivf_fp->f_pos));
-
-        session->data_fp = filp_open(data_file, O_RDONLY, 0644);
-        if (IS_ERR(session->data_fp)) {
-            MVX_SESSION_WARN(session, "warning: open file(%s) fail", data_file);
-            filp_close(ivf_fp, NULL);
-            return;
-        } else {
-            char *ivf_data = kmalloc(alloc_bytes, GFP_KERNEL);
-            if (ivf_data != NULL) {
-                size_t read_bytes;
-                // write ivf data from data file read
-                while (read_bytes = kernel_read(session->data_fp, ivf_data, alloc_bytes, &(session->data_fp->f_pos)), read_bytes > 0) {
-                    kernel_write(ivf_fp, ivf_data, read_bytes, &(ivf_fp->f_pos));
-                    if (read_bytes < alloc_bytes)
-                        break;
-                }
-                kfree(ivf_data);
-            }
-        }
-        filp_close(session->data_fp, NULL);
-        filp_close(ivf_fp, NULL);
-    }
-}
-
 static void send_event_error(struct mvx_session *session,
                  long error)
 {
@@ -3659,7 +3589,6 @@ int mvx_session_construct(struct mvx_session *session,
     session->coded_chroma_format = MVX_CHROMA_FORMAT_422;
     session->is_encoder = is_encoder;
     session->is_jpeg = is_jpeg;
-    session->enable_buffer_dump = enable_buffer_dump;
     session->priority = V4L2_SESSION_PRIORITY_DEFAULT;
     session->intra_ipenalty_angular = MVX_INVALID_VAL;
     session->intra_ipenalty_planar = MVX_INVALID_VAL;
@@ -3667,17 +3596,6 @@ int mvx_session_construct(struct mvx_session *session,
     session->inter_ipenalty_angular = MVX_INVALID_VAL;
     session->inter_ipenalty_planar = MVX_INVALID_VAL;
     session->inter_ipenalty_dc = MVX_INVALID_VAL;
-    if (session->enable_buffer_dump == true) {
-        char input_file[64];
-        scnprintf(input_file, sizeof(input_file) - 1,
-                    "/data/input_session_%p.bin", session);
-        session->data_fp = filp_open(input_file, O_RDWR | O_CREAT, 0644);
-        if (IS_ERR(session->data_fp)) {
-            MVX_SESSION_WARN(session, "warning: open dump file(%s) fail", input_file);
-            session->data_fp = NULL;
-        }
-    }
-
     memset(&session->color_conv_coef,0,sizeof(struct mvx_color_conv_coef));
     memset(&session->crop,0,sizeof(struct mvx_crop_cfg));
     memset(&session->seamless_target,0,sizeof(struct mvx_seamless_target));
@@ -3716,11 +3634,6 @@ void mvx_session_destruct(struct mvx_session *session)
 
     MVX_SESSION_INFO(session, "Destroy session.");
 
-    // dump input IVF header into a new file in the same dir with bitstream file
-    if (session->data_fp != NULL) {
-        filp_close(session->data_fp, NULL);
-        dump_ivf_header(session);
-    }
     release_fw_bin(session);
     mvx_mmu_destruct(&session->mmu);
     if (session->ts)
diff --git a/drivers/media/platform/cix/if/mvx_session.h b/drivers/media/platform/cix/if/mvx_session.h
--- a/drivers/media/platform/cix/if/mvx_session.h
+++ b/drivers/media/platform/cix/if/mvx_session.h
@@ -177,7 +177,6 @@ struct mvx_session_port {
     bool received_seq_param;
     bool pending_source_change_event;
     bool get_format_called;
-    unsigned int dump_count;
     enum mvx_format old_format;
     unsigned int old_pixelformat;
     bool buffer_rejected_flag;
@@ -202,44 +201,6 @@ struct mvx_session_qp {
     int max;
 };
 
-#pragma pack(push, 1)
-/**
- * struct mvx_ivf_header - IVF header data.
- * @signature:        DKIF signature.
- * @version:          IVF version.
- * @length:           Length of IVF header.
- * @fourcc:           Format fourcc.
- * @width:            Width.
- * @height:           Height.
- * @frameRate:        Framerate.
- * @timeScale:        Timescale.
- * @frameCount:       Total frame count.
- * @padding:          Padding value.
- */
-struct mvx_ivf_header {
-    uint32_t signature;
-    uint16_t version;
-    uint16_t length;
-    uint32_t fourcc;
-    uint16_t width;
-    uint16_t height;
-    uint32_t frameRate;
-    uint32_t timeScale;
-    uint32_t frameCount;
-    uint32_t padding;
-};
-
-/**
- * struct mvx_ivf_frame - IVF frame header data.
- * @size:        Byte size for current frame without header.
- * @timestamp:   Frame timestamp.
- */
-struct mvx_ivf_frame {
-    uint32_t size;
-    uint64_t timestamp;
-};
-#pragma pack(pop)
-
 struct mvx_session_format_map {
     enum mvx_format format;
     uint32_t flags;
@@ -472,8 +433,6 @@ struct mvx_session {
     uint64_t frame_count;
     uint32_t ts_index;
     struct mutex fps_mutex;
-    bool enable_buffer_dump;
-    struct file *data_fp;
     int fsf_mode;
     uint32_t priority;
     uint32_t intra_ipenalty_angular;
diff --git a/drivers/media/platform/cix/if/v4l2/mvx_v4l2_vidioc.c b/drivers/media/platform/cix/if/v4l2/mvx_v4l2_vidioc.c
--- a/drivers/media/platform/cix/if/v4l2/mvx_v4l2_vidioc.c
+++ b/drivers/media/platform/cix/if/v4l2/mvx_v4l2_vidioc.c
@@ -258,32 +258,6 @@ static void print_format(struct mvx_session *session,
     }
 }
 
-/**
- * dump_input_data_to_local() - dump input buffer.
- * @session:        Pointer to MVX session.
- * @vb:             Buffer to dump.
- */
-static void dump_input_data_to_local(struct mvx_session *session,
-             struct vb2_buffer *vb)
-{
-    struct mvx_session_port *input = &session->port[MVX_DIR_INPUT];
-    bool is_ivf = (input->format == MVX_FORMAT_AV1 ||
-                 input->format == MVX_FORMAT_VP8 ||
-                 input->format == MVX_FORMAT_VP9);
-    void *buffer = (void *)vb2_plane_vaddr(vb, 0);
-    unsigned int size = vb->planes[0].bytesused;
-
-    if (is_ivf) {
-        struct mvx_ivf_frame ivf_frame = {
-            .size = size,
-            .timestamp = input->dump_count
-        };
-        uint8_t *ivf_frame_head = (uint8_t*)(&ivf_frame);
-        kernel_write(session->data_fp, ivf_frame_head, sizeof(struct mvx_ivf_frame), &(session->data_fp->f_pos));
-    }
-    kernel_write(session->data_fp, buffer, size, &(session->data_fp->f_pos));
-}
-
 /**
  * queue_setup() - Initialize or verify queue parameters.
  * @q:        Videobuf2 queue.
@@ -589,10 +563,6 @@ static void buf_queue(struct vb2_buffer *b)
     if (ret != 0) {
         goto failed;
     }
-    if (dir == MVX_DIR_INPUT && session->data_fp != NULL) {
-        dump_input_data_to_local(session, b);
-        session->port[dir].dump_count++;
-    }
     ret = mvx_session_qbuf(&vsession->session, dir, &vbuf->buf);
     if (ret != 0) {
         goto failed;
diff --git a/drivers/media/platform/cix/mvx_driver.c b/drivers/media/platform/cix/mvx_driver.c
--- a/drivers/media/platform/cix/mvx_driver.c
+++ b/drivers/media/platform/cix/mvx_driver.c
@@ -39,7 +39,6 @@
 MODULE_LICENSE("GPL");
 MODULE_AUTHOR("ARMChina");
 MODULE_DESCRIPTION("CIX/ArmChina Linlon MVX VPU codec driver (amvx).");
-MODULE_IMPORT_NS("VFS_internal_I_am_really_a_filesystem_and_am_NOT_a_driver");
 MODULE_IMPORT_NS("DMA_BUF");
 
 static int __init mvx_init(void)
