--- src/pam_yubikey.c.dist
+++ src/pam_yubikey.c
@@ -38,6 +38,7 @@
 #include <stdlib.h>
 #include <stdarg.h>
 #include <string.h>
+#include <sys/resource.h>
 
 #include "libyubipam.h"
 
--- src/utils/ykvalidate.c.dist
+++ src/utils/ykvalidate.c
@@ -32,6 +32,7 @@
 #include <string.h>
 #include <unistd.h>
 #include <termios.h>
+#include <sys/resource.h>
 
 #include "ykvalidate.h"
 #include "libyubipam.h"
