diff --git a/drivers/regulator/fwnode_regulator.c b/drivers/regulator/fwnode_regulator.c
--- a/drivers/regulator/fwnode_regulator.c
+++ b/drivers/regulator/fwnode_regulator.c
@@ -93,7 +93,7 @@ static int fwnode_get_regulation_constraints(struct device *dev,
 
 	n_phandles = fwnode_count_reference_with_args(np, "regulator-coupled-with",
 						NULL);
-	if (IS_ERR(n_phandles))
+	if (n_phandles < 0)
 		n_phandles = 0;
 
 	n_phandles = max(n_phandles, 0);
