--- a/src/java-config-2.sh.in
+++ b/src/java-config-2.sh.in
@@ -9,12 +9,18 @@
 # See https://bugs.gentoo.org/show_bug.cgi?id=169925
 # for more details"
 
-if [ -z "${UID}" ] ; then
+if [ -n "${UID:-}" ] ; then
+	user_id="${UID}"
+else
 	# id lives in /usr/bin which might not be mounted
 	if type id >/dev/null 2>/dev/null ; then
 		user_id=$(id -u)
 	else
-		[ "${USER}" = "root" ] && user_id=0
+		if [ "${USER:-}" = 'root' ]; then
+			user_id=0
+		else
+			user_id=65534
+		fi
 	fi
 fi
 
