diff --git a/docs/source/markdown/podman.1.md b/docs/source/markdown/podman.1.md
index 7a6a51c399f..5755b45ac62 100644
--- a/docs/source/markdown/podman.1.md
+++ b/docs/source/markdown/podman.1.md
@@ -148,7 +148,7 @@ specify additional options via the `--storage-opt` flag.
 
 #### **\-\-storage-opt**=*value*
 
-Storage driver option, Default storage driver options are configured in /etc/containers/storage.conf (`$HOME/.config/containers/storage.conf` in rootless mode). The `STORAGE_OPTS` environment variable overrides the default. The --storage-opt specified options overrides all.
+Storage driver option, Default storage driver options are configured in /etc/containers/storage.conf (`$HOME/.config/containers/storage.conf` in rootless mode). The `STORAGE_OPTS` environment variable overrides the default. The --storage-opt specified options overrides all. If you specify --storage-opt="", no storage options will be used.
 
 #### **\-\-syslog**=*true|false*
 
diff --git a/libpod/options.go b/libpod/options.go
index 24e9d74f455..333a7c4a509 100644
--- a/libpod/options.go
+++ b/libpod/options.go
@@ -77,8 +77,7 @@ func WithStorageConfig(config storage.StoreOptions) RuntimeOption {
 				rt.storageConfig.GraphDriverOptions = make([]string, len(config.GraphDriverOptions))
 				copy(rt.storageConfig.GraphDriverOptions, config.GraphDriverOptions)
 			} else {
-				// append new options after what is specified in the config files
-				rt.storageConfig.GraphDriverOptions = append(rt.storageConfig.GraphDriverOptions, config.GraphDriverOptions...)
+				rt.storageConfig.GraphDriverOptions = config.GraphDriverOptions
 			}
 			setField = true
 		}
diff --git a/pkg/domain/infra/runtime_libpod.go b/pkg/domain/infra/runtime_libpod.go
index b0d9dc797ca..a98c9168a98 100644
--- a/pkg/domain/infra/runtime_libpod.go
+++ b/pkg/domain/infra/runtime_libpod.go
@@ -146,7 +146,11 @@ func getRuntime(ctx context.Context, fs *flag.FlagSet, opts *engineOpts) (*libpo
 	// This should always be checked after storage-driver is checked
 	if len(cfg.StorageOpts) > 0 {
 		storageSet = true
-		storageOpts.GraphDriverOptions = cfg.StorageOpts
+		if len(cfg.StorageOpts) == 1 && cfg.StorageOpts[0] == "" {
+			storageOpts.GraphDriverOptions = []string{}
+		} else {
+			storageOpts.GraphDriverOptions = cfg.StorageOpts
+		}
 	}
 	if opts.migrate {
 		options = append(options, libpod.WithMigrate())
diff --git a/test/system/005-info.bats b/test/system/005-info.bats
index 7452c19017b..c0af2e93763 100644
--- a/test/system/005-info.bats
+++ b/test/system/005-info.bats
@@ -53,4 +53,13 @@ store.imageStore.number   | 1
 
 }
 
+@test "podman info --storage-opt='' " {
+    skip_if_remote "--storage-opt flag is not supported for remote"
+    skip_if_rootless "storage opts are required for rootless running"
+    run_podman --storage-opt='' info
+    # Note this will not work in rootless mode, unless you specify
+    # storage-driver=vfs, until we have kernels that support rootless overlay
+    # mounts.
+    is "$output" ".*graphOptions: {}" "output includes graphOptions: {}"
+}
 # vim: filetype=sh
