--- /usr/portage/sys-fs/cryptsetup/cryptsetup-1.4.1.ebuild	2012-10-21 15:01:29.000000000 +0100
+++ cryptsetup-1.4.1.ebuild	2012-10-21 16:16:21.924893593 +0100
@@ -14,7 +14,7 @@
 LICENSE="GPL-2"
 SLOT="0"
 KEYWORDS="alpha amd64 arm hppa ia64 ~mips ppc ppc64 s390 sh sparc x86"
-IUSE="nls selinux +static"
+IUSE="nls selinux +static udev"
 
 S=${WORKDIR}/${MY_P}
 
@@ -26,7 +26,7 @@
 		>=sys-apps/util-linux-2.17.2
 		>=sys-fs/lvm2-2.02.64
 	)
-	>=sys-fs/udev-124
+	udev? ( >=sys-fs/udev-124 )
 	>=sys-libs/e2fsprogs-libs-1.41
 	selinux? ( sys-libs/libselinux )
 	!sys-fs/cryptsetup-luks"
@@ -37,7 +37,7 @@
 		|| ( >=sys-apps/util-linux-2.20[static-libs] <sys-apps/util-linux-2.20 )
 		dev-libs/libgcrypt[static-libs]
 		|| ( >=sys-fs/lvm2-2.02.88[static-libs] <sys-fs/lvm2-2.02.88 )
-		|| ( >=sys-fs/udev-182[static-libs] <=sys-fs/udev-171-r8 =sys-fs/udev-171* )
+		|| ( >=sys-fs/udev-182[static-libs] <=sys-fs/udev-171-r6 )
 	)"
 
 pkg_setup() {
@@ -54,12 +54,19 @@
 }
 
 src_configure() {
+	local myconf
+
+	if ! use udev; then
+		myconf="${myconf} --disable-udev"
+	fi
+
 	econf \
 		--sbindir=/sbin \
 		--enable-shared \
 		$(use_enable static static-cryptsetup) \
 		$(use_enable nls) \
-		$(use_enable selinux)
+		$(use_enable selinux) \
+		${myconf}
 }
 
 src_test() {
