--- /usr/portage/sys-fs/cryptsetup/cryptsetup-1.4.3.ebuild	2012-11-21 13:01:30.000000000 +0000
+++ cryptsetup-1.4.3.ebuild	2012-11-22 23:57:09.204324912 +0000
@@ -14,7 +14,7 @@
 LICENSE="GPL-2"
 SLOT="0"
 KEYWORDS="~alpha amd64 arm hppa ~ia64 ~mips ppc ppc64 ~s390 ~sh ~sparc x86"
-IUSE="nls selinux static static-libs"
+IUSE="nls selinux static static-libs udev"
 
 S=${WORKDIR}/${MY_P}
 
@@ -23,7 +23,7 @@
 	>=sys-apps/util-linux-2.17.2[static-libs(+)]
 	>=dev-libs/libgcrypt-1.1.42[static-libs(+)]
 	>=sys-fs/lvm2-2.02.64[static-libs(+)]
-	>=sys-fs/udev-124[static-libs(+)]
+	udev? ( >=sys-fs/udev-124[static-libs(+)] )
 	>=sys-libs/e2fsprogs-libs-1.41[static-libs(+)]"
 # We have to always depend on ${LIB_DEPEND} rather than put behind
 # static? () because we provide a shared library which links against
@@ -50,13 +50,20 @@
 }
 
 src_configure() {
+	local myconf
+
+	if ! use udev; then
+		myconf="${myconf} --disable-udev"
+	fi
+
 	econf \
 		--sbindir=/sbin \
 		--enable-shared \
 		$(use_enable static static-cryptsetup) \
 		$(use_enable static-libs static) \
 		$(use_enable nls) \
-		$(use_enable selinux)
+		$(use_enable selinux) \
+		${myconf}
 }
 
 src_test() {
