--- gdb/darwin-nat.c.dist	2014-10-31 00:16:38.000000000 +0000
+++ gdb/darwin-nat.c	2014-10-31 00:17:25.000000000 +0000
@@ -42,7 +42,11 @@
 
 #include <sys/ptrace.h>
 #include <sys/signal.h>
-#include <machine/setjmp.h>
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_10
+# include <setjmp.h>
+#else
+# include <machine/setjmp.h>
+#endif
 #include <sys/types.h>
 #include <unistd.h>
 #include <signal.h>
