--- a/gdb/compile/compile-loc2c.c
+++ b/gdb/compile/compile-loc2c.c
@@ -87,7 +87,7 @@ compute_stack_depth_worker (int start, i
     {
       enum dwarf_location_atom op = (enum dwarf_location_atom) *op_ptr;
       uint64_t reg;
-      int64_t offset;
+      int64_t offset = 0;
       int ndx = op_ptr - base;
 
 #define SET_CHECK_DEPTH(WHERE)				\
@@ -666,8 +666,8 @@ do_compile_dwarf_expr_to_c (int indent,
   while (op_ptr < op_end)
     {
       enum dwarf_location_atom op = (enum dwarf_location_atom) *op_ptr;
-      uint64_t uoffset, reg;
-      int64_t offset;
+      uint64_t uoffset = 0, reg;
+      int64_t offset = 0;
 
       print_spaces (indent - 2, &stream);
       if (info[op_ptr - base].label)
