--- gdb/configure.dist	2014-10-31 00:33:32.000000000 +0000
+++ gdb/configure	2014-10-31 00:39:24.000000000 +0000
@@ -13069,7 +13069,7 @@ $as_echo_n "checking compiler warning fl
 	-Werr*) WERROR_CFLAGS=-Werror ;;
 	*) # Check that GCC accepts it
 	    saved_CFLAGS="$CFLAGS"
-	    CFLAGS="$CFLAGS $w"
+	    CFLAGS="$CFLAGS -Werror=unknown-warning-option $w"
 	    cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
