--- gdb/configure.dist	2015-09-02 03:26:05.000000000 +0100
+++ gdb/configure	2015-09-02 03:27:11.000000000 +0100
@@ -13922,9 +13922,9 @@ $as_echo_n "checking compiler warning fl
 	*)
 	    # Check whether GCC accepts it.
 	    saved_CFLAGS="$CFLAGS"
-	    CFLAGS="$CFLAGS $w"
+	    CFLAGS="$CFLAGS -Werror=unknown-warning-option $w"
 	    saved_CXXFLAGS="$CXXFLAGS"
-	    CXXFLAGS="$CXXFLAGS $w"
+	    CXXFLAGS="$CXXFLAGS -Werror=unknown-warning-option $w"
 
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
